diff --git a/bashrc/append_num.sh b/bashrc/append_num.sh
new file mode 100644
index 00000000..d24ea5f8
--- /dev/null
+++ ./bashrc/append_num.sh
@@ -0,0 +1,3 @@
+#!/bin/zsh
+
+echo "$1" | sed "s/\/[6-9]\s\([^\/]*$\)/\/\1/g;s/\/\([^\/]*$\)/\/$2 \1/g;s/  \+/ /g;s/\"//g"
diff --git a/bashrc/append_rate.sh b/bashrc/append_rate.sh
new file mode 100644
index 00000000..7a42f1a1
--- /dev/null
+++ ./bashrc/append_rate.sh
@@ -0,0 +1,5 @@
+#!/bin/zsh
+
+source=$(echo "$1" | sed 's/"//g')
+target=$(~/loadrc/bashrc/append_num.sh "$1")
+mv -nv "$source" "$target"
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 760021ef..49fab7f4 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -1,3 +1,4 @@
+command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject AppendRate :execute s:AppendRate(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject ApplyBranch :execute s:ApplyBranch(<q-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject BinaryGrep :execute s:BinaryGrep(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject CatDu :execute s:CatDu(<f-args>)
@@ -575,6 +576,27 @@ function! s:CatMove(...) abort
     call asyncrun#run('<bang>', '', '~/loadrc/bashrc/update_proj.sh') 
 endfunction
 
+function! s:AppendRate(...) abort
+    if a:0 >= 1
+        let curword = GetWord()
+        let b:netrw_curdir = getcwd()
+        let map_escape = "<|\n\r\\\<C-V>\""
+        let mapsafecurdir = escape(b:netrw_curdir, map_escape)
+        let oldname = ComposePath(mapsafecurdir, curword)
+
+        if filereadable(oldname)
+            exec '!~/loadrc/bashrc/append_rate.sh ' . '"' .  oldname . '"' . ' ' . '"' . a:1 . '"'
+            let newname = substitute(system("~/loadrc/bashrc/append_num.sh " . '"' . oldname . '"'), '\n', '', '')
+            call setline('.', '"' . newname . '"')
+            w!
+            call UpdateProj()
+        endif
+
+    endif
+
+    call asyncrun#run('<bang>', '', '~/loadrc/bashrc/update_proj.sh')
+endfunction
+
 function! s:CatRate(...) abort
     if &modified
         echom 'Please check and save your file first!!!'
diff --git a/vishrc/cat_rate.sh b/vishrc/cat_rate.sh
index 5ace0203..e0e09066 100755
--- ./vishrc/cat_rate.sh
+++ ./vishrc/cat_rate.sh
@@ -2,7 +2,5 @@
 
 while read ss
 do
-    source=$(echo "$ss" | sed 's/"//g')
-    target=$(echo "$ss" | sed "s/\/[6-9]\s\([^\/]*$\)/\/\1/g;s/\/\([^\/]*$\)/\/$2 \1/g;s/  \+/ /g;s/\"//g")
-    mv -nv "$source" "$target"
+    ~/loadrc/bashrc/append_rate.sh "$ss"
 done < "$1"
