diff --git a/gitrc/.gitconfig b/gitrc/.gitconfig
index a0a7e0e3..ca91dbaf 100755
--- ./gitrc/.gitconfig
+++ ./gitrc/.gitconfig
@@ -54,8 +54,7 @@
 [pull]
     default = current
 [gdif]
-    ignore = *.gdio.diff
-    ignore = *.gdit.diff
+    ignore = *.diff
     ignore = .gitconfig
     ignore = .login
     ignore = COMMIT_EDITMSG
diff --git a/gitrc/apply_branch.sh b/gitrc/apply_branch.sh
new file mode 100755
index 00000000..39462a11
--- /dev/null
+++ ./gitrc/apply_branch.sh
@@ -0,0 +1,17 @@
+#!/bin/zsh
+BRANCH="$1"
+BRANCH=$(echo "$BRANCH" | sed 's/remotes\///g')
+BRANCH_DIFF=$(echo "$BRANCH" | sed 's/[^\/]*\///g')
+BRANCH_DIFF="$BRANCH_DIFF.gdio.diff"
+echo "BRANCH --> $BRANCH"
+echo "BRANCH_DIFF --> $BRANCH_DIFF"
+
+git show "$BRANCH":"$BRANCH_DIFF" > "$BRANCH_DIFF"
+retVal=$?
+
+if [ $retVal -ne 0 ]
+then
+    exit $retVal
+fi
+
+~/loadrc/gitrc/reapply.sh "$BRANCH_DIFF"
diff --git a/gitrc/apply_branch_test.sh b/gitrc/apply_branch_test.sh
new file mode 100755
index 00000000..a0e136b0
--- /dev/null
+++ ./gitrc/apply_branch_test.sh
@@ -0,0 +1,7 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+cd ../
+~/loadrc/gitrc/apply_branch.sh remotes/github/generate_files_proj
diff --git a/gitrc/git_diff_command.sh b/gitrc/git_diff_command.sh
index 601480d6..e2e8aa4b 100755
--- ./gitrc/git_diff_command.sh
+++ ./gitrc/git_diff_command.sh
@@ -1,3 +1,3 @@
 #!/bin/zsh
 
-echo "git diff --ignore-space-at-eol -b -w --ignore-blank-lines"
+echo "git diff --ignore-space-at-eol -b -w --ignore-blank-lines --binary"
diff --git a/gitrc/reapply.sh b/gitrc/reapply.sh
index 978f4284..d1ffbd95 100755
--- ./gitrc/reapply.sh
+++ ./gitrc/reapply.sh
@@ -1,7 +1,12 @@
 #!/bin/zsh
 DIFF="$1"
-git apply --reverse "$DIFF" 
-git checkout -- "$DIFF"
+CURRET_BRANCH_DIFF="$(~/loadrc/gitrc/get_current_branch.sh).gdio.diff"
+git apply --reverse "$CURRET_BRANCH_DIFF" 
+git checkout -- "$CURRET_BRANCH_DIFF"
 git apply --reject --whitespace=fix "$DIFF"
 ~/loadrc/gitrc/gdio.sh
-git add "$DIFF"
+
+if [ "$CURRET_BRANCH_DIFF" = "$DIFF" ]
+then
+    git add "$CURRET_BRANCH_DIFF"
+fi
diff --git a/prunefile_template.conf b/prunefile_template.conf
index 2154bcb9..8e34f2fa 100644
--- ./prunefile_template.conf
+++ ./prunefile_template.conf
@@ -3,3 +3,4 @@
 "./running.lck"
 "./running.time"
 .err
+"./gbra.log"
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index e30c0539..bff6d354 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -679,7 +679,6 @@ function! s:Reapply() abort
     let worktree = Cd2Worktree()
     exec '!~/loadrc/gitrc/reapply.sh ' . '"' .  expand("%:p") . '"'
     call s:Gs()
-    call s:Gdi()
 endfunction
 
 function! s:Split() abort
