diff --git a/gitrc/apply_branch.sh b/gitrc/apply_branch.sh
index 39462a11..7573f457 100755
--- ./gitrc/apply_branch.sh
+++ ./gitrc/apply_branch.sh
@@ -15,3 +15,4 @@ then
 fi
 
 ~/loadrc/gitrc/reapply.sh "$BRANCH_DIFF"
+~/loadrc/gitrc/checkout_rejs.sh "$BRANCH"
diff --git a/gitrc/checkout_rejs.sh b/gitrc/checkout_rejs.sh
index a687efd9..28171289 100755
--- ./gitrc/checkout_rejs.sh
+++ ./gitrc/checkout_rejs.sh
@@ -2,6 +2,6 @@
 for ss in $(git status | grep \.rej$) ; \
 do \
     targetFile=$(echo "$ss" | sed 's/\.rej$//g')
-    git checkout "$1" "$targetFile" && \
+    git show "$1":"$targetFile" > "$targetFile" && \
         rm "$ss"
 done
diff --git a/vimrc/.vim/bundle/swift.vim b/vimrc/.vim/bundle/swift.vim
index bfc662d1..74af6626 160000
--- ./vimrc/.vim/bundle/swift.vim
+++ ./vimrc/.vim/bundle/swift.vim
@@ -1 +1 @@
-Subproject commit bfc662d11b3a006815d0b6972c1c0c2ebbc3b16e
+Subproject commit 74af6626f63e331b60524d20dd69c9c5d03dc90c
