diff --git a/bashrc/jdupes.sh b/bashrc/jdupes.sh
index ccc72429..0527d6b9 100755
--- ./bashrc/jdupes.sh
+++ ./bashrc/jdupes.sh
@@ -22,5 +22,3 @@ then \
         ~/loadrc/bashrc/sleep.sh ; \
         fi ; \
         nvim ~/loadrc/jdupes.log
-
-echo check > /sys/block/md0/md/sync_action
diff --git a/vimrc/.vim/bundle/ncm2 b/vimrc/.vim/bundle/ncm2
index 7d49f7b4..3de17695 160000
--- ./vimrc/.vim/bundle/ncm2
+++ ./vimrc/.vim/bundle/ncm2
@@ -1 +1 @@
-Subproject commit 7d49f7b473db732878091f2f8c68bc47b015559b
+Subproject commit 3de17695c58f309642c5148b21d47fac61937a96
diff --git a/vimrc/.vim/plugin/buffergator.vim b/vimrc/.vim/plugin/buffergator.vim
new file mode 100644
index 00000000..676d1e68
--- /dev/null
+++ ./vimrc/.vim/plugin/buffergator.vim
@@ -0,0 +1,3 @@
+let g:buffergator_sort_regime = "mru"
+let g:buffergator_suppress_keymaps = 1
+nnoremap <silent> mm :BuffergatorOpen<CR>
diff --git a/vimrc/.vim/plugin/common.vim b/vimrc/.vim/plugin/common.vim
index 87bbd96d..e79dcd57 100644
--- ./vimrc/.vim/plugin/common.vim
+++ ./vimrc/.vim/plugin/common.vim
@@ -1,14 +1,10 @@
 function! CHANGE_CURR_DIR()
-    if (&filetype ==# 'fugitiveblame')
-        return
-    endif
-
     let _dir = expand("%:p:h")
 
     try
         exec "cd " . fnameescape(_dir)
     catch /.*/
-        echom 'Cought anything: ' . v:exception
+        return
     endtry
 
     unlet _dir
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 7f6cbd82..f95eb227 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -370,7 +370,7 @@ nnoremap <leader>l :TlistClose<CR>:TlistToggle<cr>
 nnoremap <leader>L :TlistClose<cr>
 nnoremap hh <c-w>h
 nnoremap ll <c-w>l
-nnoremap mm :call SwitchWinSize()<cr> 
+nnoremap mn :call SwitchWinSize()<cr> 
 nnoremap ff <c-f>
 nnoremap vv <c-b>
 nnoremap <c-l> l
diff --git a/vimrc/.vimrc b/vimrc/.vimrc
index e0b2d631..be66a22b 100644
--- ./vimrc/.vimrc
+++ ./vimrc/.vimrc
@@ -265,7 +265,6 @@ call ncm2#override_source('otherbuf', {'priority': 6})
 " set autochdir
 
 let g:indexed_search_dont_move = 1
-let g:buffergator_sort_regime = "mru"
 
 " autocmd OptionSet diff call s:ToggleOnDiff()
 " 
@@ -276,3 +275,6 @@ let g:buffergator_sort_regime = "mru"
 " endfunction
 
 let g:tex_flavor = 'latex'
+
+" add @ to cfile handle
+set isfname+=@-@
