diff --git a/.gitmodules b/.gitmodules
index 189c289f..e17f9887 100644
--- ./.gitmodules
+++ ./.gitmodules
@@ -125,10 +125,6 @@
     path = vimrc/.vim/bundle/ncm2-cssomni
     url = https://github.com/ncm2/ncm2-cssomni
     branch = master
-[submodule "vimrc/.vim/bundle/ncm2-tern"]
-    path = vimrc/.vim/bundle/ncm2-tern
-    url = https://github.com/ncm2/ncm2-tern
-    branch = master
 [submodule "vimrc/.vim/bundle/ncm2-jedi"]
     path = vimrc/.vim/bundle/ncm2-jedi
     url = https://github.com/ncm2/ncm2-jedi
diff --git a/vimrc/.vim/bundle/ncm2-tern b/vimrc/.vim/bundle/ncm2-tern
deleted file mode 160000
index 04febdd4..00000000
--- ./vimrc/.vim/bundle/ncm2-tern
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 04febdd474ca586fff16accb1b5b268c2dde282e
