diff --git a/imvurc/generate.sh b/imvurc/generate.sh
index b32db637..7c49d05c 100755
--- ./imvurc/generate.sh
+++ ./imvurc/generate.sh
@@ -5,7 +5,7 @@ target="$1.sh"
 if [ ! -f "$target" ]
 then
     cp -nv ~/loadrc/imvurc/generate_sh.template "$target"
-    rootFolder=$(~/loadrc/bashrc/find_up_folder.sh "$file" "files.proj")
+    rootFolder=$(~/loadrc/bashrc/find_up_folder.sh "$PWD/$file" "files.proj")
     relative_path=$(realpath --relative-to="$rootFolder" "$file")
     relative_path=$(echo "$relative_path" | sed 's/\//\\\//g')
     sed -i.bak "s/relative_path_var/$relative_path/g" "$target"
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index f5d17f68..ef070305 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit f5d17f68a016946155e728752d3ced0359ab909c
+Subproject commit ef0703052f64ce26a38d00ac64b86d0260da74e2
