diff --git a/bashrc/select_move.sh b/bashrc/select_move.sh
new file mode 100755
index 00000000..29fd5336
--- /dev/null
+++ ./bashrc/select_move.sh
@@ -0,0 +1,8 @@
+#!/bin/zsh
+
+cat "$1" | sed 's/^"//g;s/"$//g;s/\\ / /g' > "$1".tmp
+rsync -aHv --force --progress \
+    --remove-source-files \
+    --files-from "$1".tmp \
+    "$2" \
+    "$3"
diff --git a/gitrc/gcom.sh b/gitrc/gcom.sh
index 83a83235..e8b5bfa5 100755
--- ./gitrc/gcom.sh
+++ ./gitrc/gcom.sh
@@ -25,3 +25,4 @@ then
 fi
 
 ~/loadrc/gitrc/gdio.sh
+git checkout "$CURRENT_BRANCH.gdio.diff"
diff --git a/imvurc/ghypo.sh b/imvurc/ghypo.sh
index f04e3bfa..bd703fed 100755
--- ./imvurc/ghypo.sh
+++ ./imvurc/ghypo.sh
@@ -26,5 +26,6 @@ else
     if [ $(~/loadrc/gitrc/git_ready.sh) ]
     then
         s/hypo
+        #s/hypo --buildbot=hypo2 
     fi
 fi
diff --git a/install_prerequisite.sh b/install_prerequisite.sh
index f055c7a2..cd4e7524 100755
--- ./install_prerequisite.sh
+++ ./install_prerequisite.sh
@@ -29,6 +29,7 @@ apt-get install -y \
     iperf \
     iperf3 \
     iputils-ping \
+    jq \
     kdiff3 \
     libsecret-1-dev \
     libsecret-tools \
@@ -36,6 +37,7 @@ apt-get install -y \
     manpages-fr-extra \
     manpages-pl \
     mdadm \
+    moreutils \
     mplayer \
     net-tools \
     netcat \
diff --git a/neovim b/neovim
index b0ed4793..5b9dc8f8 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit b0ed4793be6df3e1f951fd1874fbc5b3e266c91b
+Subproject commit 5b9dc8f8f802bda44e091bc20845274019b25e05-dirty
diff --git a/vimrc/.vim/bundle/neco-vim b/vimrc/.vim/bundle/neco-vim
index 31e5725a..cd666d51 160000
--- ./vimrc/.vim/bundle/neco-vim
+++ ./vimrc/.vim/bundle/neco-vim
@@ -1 +1 @@
-Subproject commit 31e5725ab5666396921d86bcec2f0b6ea0028ce7
+Subproject commit cd666d51cb4d03e509039fd35e5744bbf9db431b
diff --git a/vimrc/.vim/bundle/swift.vim b/vimrc/.vim/bundle/swift.vim
index e83ca052..3278cf3b 160000
--- ./vimrc/.vim/bundle/swift.vim
+++ ./vimrc/.vim/bundle/swift.vim
@@ -1 +1 @@
-Subproject commit e83ca052e88805b8b20bc2558e7995461665e306
+Subproject commit 3278cf3b0522e6f08eaf11275fedce619beffe9a
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index c395bf31..1bb095f1 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit c395bf31769baf4ccaf4d6e255038c375a1672bb
+Subproject commit 1bb095f1cd592b5e381f89ae875c8689530f1eb0
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 76b1a4eb..040d8912 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 76b1a4ebcb7ec8aba88fd3072649f13b1a783585
+Subproject commit 040d89120c4910586828b42963307061567f9431-dirty
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 029b616a..b750ffe4 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -3,6 +3,7 @@ command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Binary
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject CatDu :execute s:CatDu(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject CatMove :execute s:CatMove(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject CatPlay :execute s:CatPlay(<f-args>)
+command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject CatRate :execute s:CatRate(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject CatRun :execute s:CatRun(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Copy :execute s:Copy(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Dodev :execute s:Dodev()
@@ -538,6 +539,11 @@ function! s:Gsubbr() abort
 endfunction
 
 function! s:Copy(...) abort
+    if &modified
+        echom 'Please check and save your file first!!!'
+        return 0
+    endif
+
     let newFile = (a:0 >= 1) ? a:1 : expand("%:p") . '.bak'
     exec '!rm ' . newFile
     exec 'w ' . newFile
@@ -569,6 +575,19 @@ function! s:CatMove(...) abort
     call asyncrun#run('<bang>', '', '~/loadrc/bashrc/update_proj.sh') 
 endfunction
 
+function! s:CatRate(...) abort
+    if &modified
+        echom 'Please check and save your file first!!!'
+        return 0
+    endif
+
+    if a:0 >= 1
+        exec '!~/loadrc/vishrc/cat_rate.sh ' . '"' .  expand("%:p") . '"' . ' ' . '"' . a:1 . '"'
+    endif
+
+    call asyncrun#run('<bang>', '', '~/loadrc/bashrc/update_proj.sh') 
+endfunction
+
 function! s:CatDu(...) abort
     if &modified
         echom 'Please check and save your file first!!!'
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 9a974dbd..12788667 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -77,8 +77,16 @@ function! VFilter()
 endfunction
 
 function! ShowRemember()
-    let @" = expand('%:p')
-    echom expand('%:p')
+    let b:csdbpath = Cd2ProjectRoot("files.proj")
+    let relativePath = substitute(system('realpath --relative-to="' . b:csdbpath . '" ' . expand('%:p')), '\n', '', '')
+
+    if &clipboard == 'unnamed'
+        let @* = relativePath
+    else
+        let @+ = relativePath
+    endif
+
+    echom relativePath
     call SendTextToPbCopy(expand('%:p'))
 endfunction
 
diff --git a/vishrc/cat_rate.sh b/vishrc/cat_rate.sh
new file mode 100755
index 00000000..5ace0203
--- /dev/null
+++ ./vishrc/cat_rate.sh
@@ -0,0 +1,8 @@
+#!/bin/zsh
+
+while read ss
+do
+    source=$(echo "$ss" | sed 's/"//g')
+    target=$(echo "$ss" | sed "s/\/[6-9]\s\([^\/]*$\)/\/\1/g;s/\/\([^\/]*$\)/\/$2 \1/g;s/  \+/ /g;s/\"//g")
+    mv -nv "$source" "$target"
+done < "$1"
