diff --git a/bashrc/select_move.sh b/bashrc/select_move.sh
new file mode 100755
index 00000000..29fd5336
--- /dev/null
+++ ./bashrc/select_move.sh
@@ -0,0 +1,8 @@
+#!/bin/zsh
+
+cat "$1" | sed 's/^"//g;s/"$//g;s/\\ / /g' > "$1".tmp
+rsync -aHv --force --progress \
+    --remove-source-files \
+    --files-from "$1".tmp \
+    "$2" \
+    "$3"
diff --git a/neovim b/neovim
index b0ed4793..e57dd23d 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit b0ed4793be6df3e1f951fd1874fbc5b3e266c91b
+Subproject commit e57dd23df401637a48325f39bc655c8bff30dadd
diff --git a/vimrc/.vim/bundle/neco-vim b/vimrc/.vim/bundle/neco-vim
index 31e5725a..cd666d51 160000
--- ./vimrc/.vim/bundle/neco-vim
+++ ./vimrc/.vim/bundle/neco-vim
@@ -1 +1 @@
-Subproject commit 31e5725ab5666396921d86bcec2f0b6ea0028ce7
+Subproject commit cd666d51cb4d03e509039fd35e5744bbf9db431b
diff --git a/vimrc/.vim/bundle/swift.vim b/vimrc/.vim/bundle/swift.vim
index e83ca052..3278cf3b 160000
--- ./vimrc/.vim/bundle/swift.vim
+++ ./vimrc/.vim/bundle/swift.vim
@@ -1 +1 @@
-Subproject commit e83ca052e88805b8b20bc2558e7995461665e306
+Subproject commit 3278cf3b0522e6f08eaf11275fedce619beffe9a
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index c395bf31..1bb095f1 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit c395bf31769baf4ccaf4d6e255038c375a1672bb
+Subproject commit 1bb095f1cd592b5e381f89ae875c8689530f1eb0
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 76b1a4eb..9aa4231b 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 76b1a4ebcb7ec8aba88fd3072649f13b1a783585
+Subproject commit 9aa4231b1120af08e263fc88210bd4a731585463
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 029b616a..42dfdc56 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -538,6 +538,11 @@ function! s:Gsubbr() abort
 endfunction
 
 function! s:Copy(...) abort
+    if &modified
+        echom 'Please check and save your file first!!!'
+        return 0
+    endif
+
     let newFile = (a:0 >= 1) ? a:1 : expand("%:p") . '.bak'
     exec '!rm ' . newFile
     exec 'w ' . newFile
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 9a974dbd..12788667 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -77,8 +77,16 @@ function! VFilter()
 endfunction
 
 function! ShowRemember()
-    let @" = expand('%:p')
-    echom expand('%:p')
+    let b:csdbpath = Cd2ProjectRoot("files.proj")
+    let relativePath = substitute(system('realpath --relative-to="' . b:csdbpath . '" ' . expand('%:p')), '\n', '', '')
+
+    if &clipboard == 'unnamed'
+        let @* = relativePath
+    else
+        let @+ = relativePath
+    endif
+
+    echom relativePath
     call SendTextToPbCopy(expand('%:p'))
 endfunction
 
