diff --git a/.gitconfig_sample b/.gitconfig_sample
index 639febb9..0ea38ff6 100644
--- ./.gitconfig_sample
+++ ./.gitconfig_sample
@@ -8,7 +8,7 @@
     remote = origin
 [gsync]
     remote = authorVar
-    target = authorVar/master
+    target = authorVar/branchVar
 [merge]
     allow = true
 [branch "dev"]
diff --git a/gitrc/gcl.sh b/gitrc/gcl.sh
index e118a4de..e1f58816 100755
--- ./gitrc/gcl.sh
+++ ./gitrc/gcl.sh
@@ -18,7 +18,10 @@ urlVar=$(echo "$urlVar" | sed 's/\//\\\//g')
 sed -i.bak "s|urlVar|$urlVar|g;s|repoVar|${author}_${repo}|g;s|authorVar|$author|g" "$target"/.gitconfig
 echo ".gitconfig --> $(realpath ${target}/.gitconfig)"
 
-cd "$target" && \
+cd "$target"
+CURRENT_BRANCH=$(~/loadrc/gitrc/get_current_branch.sh)
+sed -i.bak "s|branchVar|$CURRENT_BRANCH|g" .gitconfig
+
 git remote remove origin ; \
     $HOME/loadrc/gitrc/gcob.sh dev && \
     $HOME/loadrc/gitrc/gdev.sh && \
diff --git a/neovim b/neovim
index 2652732b..ead7890a 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit 2652732b7ee66c0a47d0dc0be3100700dbdab61b
+Subproject commit ead7890a376585c8c12f5c748bb5f1f2d52511b0
diff --git a/vimrc/.vim/bundle/neco-vim b/vimrc/.vim/bundle/neco-vim
index ba9b6535..6cbf6f06 160000
--- ./vimrc/.vim/bundle/neco-vim
+++ ./vimrc/.vim/bundle/neco-vim
@@ -1 +1 @@
-Subproject commit ba9b6535381690fc6773d682fc046d8ddd2a863a
+Subproject commit 6cbf6f0610e3c194366fc938b4a0ad572ad476e9
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index a8d0cbeb..dea740bc 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit a8d0cbebf04cfe557abefbd0604a0613f518052c
+Subproject commit dea740bc53bf8e3ee69d011b24ac24f923c69970
diff --git a/vimrc/.vim/plugin/clipboard.vim b/vimrc/.vim/plugin/clipboard.vim
deleted file mode 100644
index 6ccf639b..00000000
--- ./vimrc/.vim/plugin/clipboard.vim
+++ /dev/null
@@ -1,7 +0,0 @@
-let os = substitute(system('uname'), "\n", "", "")
-
-if os == "Linux"
-    set clipboard=unnamedplus
-else
-    set clipboard=unnamed
-endif
diff --git a/vimrc/.vim/plugin/copypath.vim b/vimrc/.vim/plugin/copypath.vim
deleted file mode 100755
index 65740cb6..00000000
--- ./vimrc/.vim/plugin/copypath.vim
+++ /dev/null
@@ -1,88 +0,0 @@
-" copypath.vim : Copy current editing file path to clipboard.
-"
-" Name Of File: copypath.vim
-" Maintainer:   omi taku <mail@nanasi.jp>
-" URL:          http://nanasi.jp/
-" Script URL:   http://www.vim.org/scripts/script.php?script_id=1456
-" Last Change:  2009/12/16
-" Version:      1.0
-"
-" Installation:
-"    1. Unzip copypath.zip ,
-"       and extract there to your vim runtimepath directory ,
-"       $HOME/vimfiles/plugin or $HOME/.vim/plugin directory directory.
-"
-"       Refer to ':help add-plugin', ':help add-global-plugin' and ':help runtimepath'
-"       for more details about Vim plugins.
-"
-"    2. Restart Vim.
-"    3. To run this script, Vim needs to be compiled with "+clipboard" option.
-"
-" Usage:
-"    This script adds two new command ":CopyPath" and ":CopyFileName".
-"    You can use the ":CopyPath" command to copy file path to clipboard,
-"    and you can use the ":CopyFileName" command to copy file name to clipboard.
-"
-"    :CopyPath
-"        copy current editing file full path to clipboard.
-"        And, if you set g:copypath_copy_to_unnamed_register option to 1 then,
-"        copy file full path to unnamed register too.
-"        Default, not copy to unnamed register.
-"
-"    :CopyFileName
-"        copy current editing file name to clipboard.
-"        And, if you set g:copypath_copy_to_unnamed_register option to 1 then,
-"        copy file name to unnamed register too.
-"        Default, not copy to unnamed register.
-"
-" Configuration:
-"    g:copypath_copy_to_unnamed_register
-"       set this option value to 1 then, copy file path and name to unnamed register too.
-"   >
-"       let g:copypath_copy_to_unnamed_register = 1
-"   <
-"
-" Note:
-"    o  To run this script, Vim needs to be compiled with "+clipboard" option.
-"
-" History:
-"    1.0     o  change uploading archive file format to zip format.
-"    0.2     o  unnamed register copy option is added.
-"    0.1.2   o  document is updated.
-"    0.1.1   o  fix bug.
-"    0.1     o  Initial upload.
-
-
-" if plugin is already loaded then, not load plugin.
-if exists("g:loaded_copypath")
-    finish
-endif
-let g:loaded_copypath = 1
-
-" if option is set then use open.
-if !exists('g:copypath_copy_to_unnamed_register')
-    let g:copypath_copy_to_unnamed_register = 0
-endif
-
-function CopyPath()
-    let @+=expand('%:p')
-    " copy unnamed register.
-    if g:copypath_copy_to_unnamed_register
-        let @"=expand('%:p')
-    endif
-endfunction
-
-function CopyFileName()
-    let @+=expand('%:t')
-    " copy unnamed register.
-    if g:copypath_copy_to_unnamed_register
-        let @"=expand('%:t')
-    endif
-endfunction
-
-command! -nargs=0 CopyPath     call CopyPath()
-command! -nargs=0 CopyFileName call CopyFileName()
-
-finish
-
-" vim: set ts=4 sts=4 sw=4 et nowrap :
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 78fb9e11..490aaa00 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -77,7 +77,7 @@ function! VFilter()
 endfunction
 
 function! ShowRemember()
-    let @+=expand('%:p')
+    let @" = expand('%:p')
     echom expand('%:p')
 endfunction
 
@@ -353,13 +353,6 @@ function! VimSearch()
     call HighlightKeyword(keyword)
 endfunction
 
-function! OpenProjectRoot()
-    let b:csdbpath = Find_in_parent("files.proj", Windowdir(), "/")
-    let @+=b:csdbpath
-    echom b:csdbpath
-    call OpenOrSwitch(b:csdbpath, 'vs')
-endfunction
-
 function! SwitchWinSize()
     if &winwidth == 1
         set winwidth=999999
@@ -418,34 +411,13 @@ nnoremap <leader>P "+P
 nnoremap tt :Autoformat<CR>:w!<cr>
 nnoremap D :only<CR>:vs %:p<cr>:set winwidth=1<cr><c-w>=
 " Quickly open current dir in current windows
-nnoremap <leader>d :call OpenProjectRoot()<cr>
 nnoremap <tab> %
 vnoremap <tab> %
 nnoremap M zM
 nnoremap R zR
 nmap <f2> :set number! number?<cr>
 nmap <leader>w :call WinDo('set wrap!') <cr>
-" Convert slashes to backslashes for Windows.
-if has('win32')
-    nmap <leader>cs :let @*=substitute(expand("%"), "/", "\\", "g")<CR>
-    nmap <leader>cl :let @*=substitute(expand("%:p"), "/", "\\", "g")<CR>
 
-    " This will copy the path in 8.3 short format, for DOS and Windows 9x
-    nmap <leader>c8 :let @*=substitute(expand("%:p:8"), "/", "\\", "g")<CR>
-else
-    nmap <leader>cs :let @*=expand("%")<CR>
-    nmap <leader>cl :let @*=expand("%:p")<CR>
-endif
-if has('win32')
-    nmap ,cs :let @*=substitute(expand("%"), "/", "\\", "g")<CR>
-    nmap ,cl :let @*=substitute(expand("%:p"), "/", "\\", "g")<CR>
-
-    " This will copy the path in 8.3 short format, for DOS and Windows 9x
-    nmap ,c8 :let @*=substitute(expand("%:p:8"), "/", "\\", "g")<CR>
-else
-    nmap fs :let @"=expand("%")<CR>
-    nmap <leader>p :let @"=expand("%:p")<CR>
-endif
 " nnoremap F :echom expand('%:p')<cr>
 vnoremap <silent>f :call VimSearch()<cr>
 vnoremap <silent>s :call GitSearch()<cr>
@@ -510,9 +482,10 @@ nnoremap <leader>T :TigOpenCurrentFile<CR>
 nnoremap <silent> ml :call CopyLineInfo()<cr> 
 
 function! CopyLineInfo()
+    normal yy
     let b:csdbpath = Cd2ProjectRoot("files.proj")
     let relativePath = substitute(system('realpath --relative-to="' . b:csdbpath . '" ' . expand('%:p')), '\n', '', '')
-    let content = relativePath . ':' . line('.') . ' ' . getline(line('.'))
-    let @+=content
+    let content = relativePath . ':' . line('.') . ' ' . @"
+    let @" = content
 endfunction
 
diff --git a/vimrc/.vim/plugin/pbpaste.vim b/vimrc/.vim/plugin/pbpaste.vim
index df39432d..8ad57466 100644
--- ./vimrc/.vim/plugin/pbpaste.vim
+++ ./vimrc/.vim/plugin/pbpaste.vim
@@ -1,9 +1,5 @@
 function! PopulatePasteBufferFromOSX(mode)
-    if &clipboard == 'unnamed'
-        let @* = system('pbpaste-remote')
-    else
-        let @+ = system('pbpaste-remote')
-    endif
+    let @" = system('pbpaste-remote')
 
     if a:mode== 'v'
         normal! gvp
