diff --git a/.macos.rc b/.macos.rc
new file mode 100644
index 00000000..9e7032fd
--- /dev/null
+++ ./.macos.rc
@@ -0,0 +1 @@
+~/loadrc/setup_pbpaste-remote.sh
diff --git a/gitrc/discard_unnecessaries.sh b/gitrc/discard_unnecessaries.sh
index de63aa2c..9df3e01a 100755
--- ./gitrc/discard_unnecessaries.sh
+++ ./gitrc/discard_unnecessaries.sh
@@ -1,3 +1,4 @@
 #!/bin/zsh
+git checkout "$(~/loadrc/gitrc/get_current_branch.sh).gdio.diff"
 git checkout files.proj
 git checkout rsync.files
