diff --git a/gitrc/g.sh b/gitrc/g.sh
index 3c402327..2477da66 100755
--- ./gitrc/g.sh
+++ ./gitrc/g.sh
@@ -1,8 +1,6 @@
 #!/bin/zsh
 ~/loadrc/gitrc/include_gitconfig.sh
 
-commit_message="n"
-
 if [ -f COMMIT_EDITMSG ]
 then
     commit_message=$(cat COMMIT_EDITMSG)
@@ -13,6 +11,11 @@ then
     commit_message="$1"
 fi
 
+if [ -z "$commit_message" ]
+then
+    commit_message="n"
+fi
+
 git commit --no-edit || \
     git commit -m "$commit_message"
 
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index dbe8b1e2..e3589389 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit dbe8b1e21e987866bde27a008deda5d0d767bee3
+Subproject commit e35893897f0f15324dd7ea2924067b228008f3ed-dirty
