diff --git a/bashrc/cscope.sh b/bashrc/cscope.sh
index 53723ae8..1a76de03 100755
--- ./bashrc/cscope.sh
+++ ./bashrc/cscope.sh
@@ -1,66 +1,6 @@
 #!/bin/zsh
 
-if [ -z "$1" ]
-then
-    TARGETEDIR=`realpath "$PWD"`
-else
-    TARGETEDIR=`realpath "$1"`
-fi
-
-cd "$TARGETEDIR"
-
-if [ ! -f files.proj ]
-then
-    echo -e "${red}No files.proj file here, will not build the index ... ${NC}"
-    exit 1
-fi
-
-TARGETEDIR=`realpath "$PWD"`
-cp -nv ~/loadrc/prunefix_template.conf ./prunefix.conf
-cp -nv ~/loadrc/prunefile_template.conf ./prunefile.conf
-cp -nv ~/loadrc/includefile_template.conf ./includefile.conf
-cp -nv ~/loadrc/setup.cfg ./setup.cfg
-TARGET=files.proj.bak
-PRUNE_POSTFIX=prunefix.conf
-PRUNE_FILE=prunefile.conf
-INCLUDE_FILE=includefile.conf
-
-or="";
-
-[ -f "$PRUNE_POSTFIX" ] && while read suf
-do
-    suf=$(echo "$suf" | sed 's/"//g')
-    prune_params+=( $or "-wholename" "$suf" )
-    or="-o"
-done < "$PRUNE_POSTFIX"
-
-or="";
-
-[ -f "$INCLUDE_FILE" ] && while read suf
-do
-    suf=$(echo "$suf" | sed 's/"//g')
-    include_params+=( $or "-wholename" "$suf" )
-    or="-o"
-done < "$INCLUDE_FILE"
-
-export LC_ALL=C
-find . "(" "${prune_params[@]}" ")" -a -prune -o -size +0 -type f -exec grep -Il "" {} + | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' > "$TARGET" && \
-    comm -23 <(sort "$TARGET") <(sort "$PRUNE_FILE") > "$TARGET.tmp" && \
-    cp -fv "$TARGET.tmp" "$TARGET" && \
-if [ ${#include_params[@]} -gt 0 ] ; \
-then \
-    find . "(" "${include_params[@]}" ")" -type f -size -9000k | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' >> ${TARGET} ; \
-    fi && \
-    sort -u "$TARGET" -o "$TARGET" && \
-    cp -fv "$TARGET" files.proj && \
-    sed -i.bak 's/ /\\ /g' files.proj && \
-    ~/loadrc/bashrc/fvideos.sh && \
-    ~/loadrc/bashrc/fdocs.sh && \
-    echo "$TARGETEDIR"/files.proj | sed 's/\(["'\''\]\)/\\\1/g;s/ /\\ /g;s/.*/"&"/' >> ~/all.proj && \
-    cscope -bq -i "$TARGET" -f cscope.out.bak && \
+cscope -bq -i files.proj -f cscope.out.bak && \
     cp -fv cscope.out.bak cscope.out && \
     cp -fv cscope.out.bak.in cscope.out.in && \
     cp -fv cscope.out.bak.po cscope.out.po
-
-sort -u ~/all.proj -o ~/all.proj
-~/loadrc/bashrc/generate_rsync_files.sh
diff --git a/bashrc/do_update_proj.sh b/bashrc/do_update_proj.sh
new file mode 100755
index 00000000..df268c7b
--- /dev/null
+++ ./bashrc/do_update_proj.sh
@@ -0,0 +1,7 @@
+#!/bin/zsh
+
+~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/generate_files_proj.sh &
+~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/fvideos.sh &
+~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/fdocs.sh &
+~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/cscope.sh &
+~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/generate_rsync_files.sh &
diff --git a/bashrc/fcscope.sh b/bashrc/fcscope.sh
index 4ca69982..ca4becd7 100755
--- ./bashrc/fcscope.sh
+++ ./bashrc/fcscope.sh
@@ -3,5 +3,5 @@ cscopes=($(find . -type f -name files.proj))
 for cscope in "${cscopes[@]}"
 do
   echo "${cscope}"
-  ~/loadrc/bashrc/cscope.sh $(dirname "${cscope}")
+  ~/loadrc/bashrc/do_update_proj.sh $(dirname "${cscope}")
 done
diff --git a/bashrc/generate_files_proj.sh b/bashrc/generate_files_proj.sh
new file mode 100755
index 00000000..5f5f2924
--- /dev/null
+++ ./bashrc/generate_files_proj.sh
@@ -0,0 +1,60 @@
+#!/bin/zsh
+
+if [ -z "$1" ]
+then
+    TARGETEDIR=`realpath "$PWD"`
+else
+    TARGETEDIR=`realpath "$1"`
+fi
+
+cd "$TARGETEDIR"
+
+if [ ! -f files.proj ]
+then
+    echo -e "${red}No files.proj file here, will not build the index ... ${NC}"
+    exit 1
+fi
+
+TARGETEDIR=`realpath "$PWD"`
+cp -nv ~/loadrc/prunefix_template.conf ./prunefix.conf
+cp -nv ~/loadrc/prunefile_template.conf ./prunefile.conf
+cp -nv ~/loadrc/includefile_template.conf ./includefile.conf
+cp -nv ~/loadrc/setup.cfg ./setup.cfg
+TARGET=files.proj.bak
+PRUNE_POSTFIX=prunefix.conf
+PRUNE_FILE=prunefile.conf
+INCLUDE_FILE=includefile.conf
+
+or="";
+
+[ -f "$PRUNE_POSTFIX" ] && while read suf
+do
+    suf=$(echo "$suf" | sed 's/"//g')
+    prune_params+=( $or "-wholename" "$suf" )
+    or="-o"
+done < "$PRUNE_POSTFIX"
+
+or="";
+
+[ -f "$INCLUDE_FILE" ] && while read suf
+do
+    suf=$(echo "$suf" | sed 's/"//g')
+    include_params+=( $or "-wholename" "$suf" )
+    or="-o"
+done < "$INCLUDE_FILE"
+
+export LC_ALL=C
+
+find . "(" "${prune_params[@]}" ")" -a -prune -o -size +0 -type f -exec grep -Il "" {} + | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' > "$TARGET" && \
+    comm -23 <(sort "$TARGET") <(sort "$PRUNE_FILE") > "$TARGET.tmp" && \
+    cp -fv "$TARGET.tmp" "$TARGET" && \
+    if [ ${#include_params[@]} -gt 0 ] ; \
+    then \
+        find . "(" "${include_params[@]}" ")" -type f -size -9000k | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' >> ${TARGET} ; \
+        fi && \
+        echo "$(git rev-parse --abbrev-ref HEAD).gdio.diff" | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' >> ${TARGET} && \
+        sort -u "$TARGET" -o "$TARGET" && \
+        sed -i.bak 's/ /\\ /g' "$TARGET" && \
+        cp -fv "$TARGET" files.proj && \
+        echo "$TARGETEDIR"/files.proj | sed 's/\(["'\''\]\)/\\\1/g;s/ /\\ /g;s/.*/"&"/' >> ~/all.proj && \
+        sort -u ~/all.proj -o ~/all.proj
diff --git a/bashrc/generate_rsync_files.sh b/bashrc/generate_rsync_files.sh
index 4477b6da..d5360ed8 100755
--- ./bashrc/generate_rsync_files.sh
+++ ./bashrc/generate_rsync_files.sh
@@ -26,9 +27,9 @@ done < "$INCLUDE_FILE"
 
 if [ ${#prune_params[@]} -gt 0 ] ; \
 then \
-    find . "(" "${prune_params[@]}" ")" -type f -size -9000k > "$rsyncFiles".diff && \
-    comm -23 <(sort "$rsyncFiles") <(sort "$rsyncFiles".diff) > "$rsyncFiles".tmp && \
-    cp -fv "$rsyncFiles".tmp "$rsyncFiles" ; \
+    find . "(" "${prune_params[@]}" ")" -type f -size -9000k > "$rsyncFiles.diff" && \
+    comm -23 <(sort "$rsyncFiles") <(sort "$rsyncFiles".diff) > "$rsyncFiles.tmp" && \
+    cp -fv "$rsyncFiles.tmp" "$rsyncFiles" ; \
     fi && \
     if [ ${#include_params[@]} -gt 0 ] ; \
     then \
diff --git a/bashrc/jdupes.sh b/bashrc/jdupes.sh
index ccc72429..0527d6b9 100755
--- ./bashrc/jdupes.sh
+++ ./bashrc/jdupes.sh
@@ -22,5 +22,3 @@ then \
         ~/loadrc/bashrc/sleep.sh ; \
         fi ; \
         nvim ~/loadrc/jdupes.log
-
-echo check > /sys/block/md0/md/sync_action
diff --git a/bashrc/update_proj.sh b/bashrc/update_proj.sh
index 6bf1ff3a..e655b2f5 100755
--- ./bashrc/update_proj.sh
+++ ./bashrc/update_proj.sh
@@ -1,3 +1,3 @@
 #!/bin/zsh
 ~/loadrc/bashrc/find_up_goto.sh "files.proj"
-~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/cscope.sh
+~/loadrc/bashrc/do_update_proj.sh
diff --git a/gitrc/gcl.sh b/gitrc/gcl.sh
index 693a3f6c..e118a4de 100755
--- ./gitrc/gcl.sh
+++ ./gitrc/gcl.sh
@@ -22,7 +22,7 @@ cd "$target" && \
     git remote remove origin ; \
     $HOME/loadrc/gitrc/gcob.sh dev && \
     $HOME/loadrc/gitrc/gdev.sh && \
-    $HOME/loadrc/bashrc/cscope.sh && \
+    $HOME/loadrc/bashrc/do_update_proj.sh && \
     git add . && \
     git add -f files.proj .gitconfig && \
     $HOME/loadrc/gitrc/g.sh && \
diff --git a/vimrc/.vim/plugin/buffergator.vim b/vimrc/.vim/plugin/buffergator.vim
new file mode 100644
index 00000000..676d1e68
--- /dev/null
+++ ./vimrc/.vim/plugin/buffergator.vim
@@ -0,0 +1,3 @@
+let g:buffergator_sort_regime = "mru"
+let g:buffergator_suppress_keymaps = 1
+nnoremap <silent> mm :BuffergatorOpen<CR>
diff --git a/vimrc/.vim/plugin/common.vim b/vimrc/.vim/plugin/common.vim
index 87bbd96d..e79dcd57 100644
--- ./vimrc/.vim/plugin/common.vim
+++ ./vimrc/.vim/plugin/common.vim
@@ -1,14 +1,10 @@
 function! CHANGE_CURR_DIR()
-    if (&filetype ==# 'fugitiveblame')
-        return
-    endif
-
     let _dir = expand("%:p:h")
 
     try
         exec "cd " . fnameescape(_dir)
     catch /.*/
-        echom 'Cought anything: ' . v:exception
+        return
     endtry
 
     unlet _dir
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 66887e96..99198791 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -645,7 +645,7 @@ function! s:Fr(find, replace) abort
 endfunction
 
 function! s:FindDeleted() abort
-    call Cd2ProjectRoot("files.proj")
+    call Cd2Worktree()
     silent exec '!~/loadrc/gitrc/find_deleted.sh 2>&1 | tee find_deleted.findresult'
     call OpenOrSwitch('find_deleted.findresult', 'vs')
 endfunction
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 7f6cbd82..0671295a 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -140,7 +140,7 @@ function! VRun()
         let b:to_run = b:to_run . '.sh'
     endif
 
-    let b:csdbpath = Cd2ProjectRoot("files.proj")
+    let b:csdbpath = Cd2Worktree()
     let b:output = b:csdbpath . '/' . b:file_name . '.runresult'
     call RunShell('~/loadrc/vishrc/vrun.sh', b:to_run, b:output)
 
@@ -370,7 +370,7 @@ nnoremap <leader>l :TlistClose<CR>:TlistToggle<cr>
 nnoremap <leader>L :TlistClose<cr>
 nnoremap hh <c-w>h
 nnoremap ll <c-w>l
-nnoremap mm :call SwitchWinSize()<cr> 
+nnoremap mn :call SwitchWinSize()<cr> 
 nnoremap ff <c-f>
 nnoremap vv <c-b>
 nnoremap <c-l> l
diff --git a/vimrc/.vimrc b/vimrc/.vimrc
index e0b2d631..be66a22b 100644
--- ./vimrc/.vimrc
+++ ./vimrc/.vimrc
@@ -265,7 +265,6 @@ call ncm2#override_source('otherbuf', {'priority': 6})
 " set autochdir
 
 let g:indexed_search_dont_move = 1
-let g:buffergator_sort_regime = "mru"
 
 " autocmd OptionSet diff call s:ToggleOnDiff()
 " 
@@ -276,3 +275,6 @@ let g:buffergator_sort_regime = "mru"
 " endfunction
 
 let g:tex_flavor = 'latex'
+
+" add @ to cfile handle
+set isfname+=@-@
