diff --git a/bashrc/cutcommon.sh b/bashrc/cutcommon.sh
new file mode 100755
index 00000000..03ad0cc4
--- /dev/null
+++ ./bashrc/cutcommon.sh
@@ -0,0 +1,7 @@
+#!/bin/zsh
+export LC_ALL=C
+
+comm -2 -3 <(sort "$1") <(sort "$2") > "$1".findresult
+comm -1 -3 <(sort "$1") <(sort "$2") > "$2".findresult 
+cp -fv "$1".findresult "$1"
+cp -fv "$2".findresult "$2"
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 12788667..76224da9 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -473,10 +473,7 @@ function! CutFile2()
 endfunc
 
 function! CutCommon()
-    silent exec '!comm -2 -3 <(sort "' . @" . '") <(sort "' . expand("%:p") . '") > "' . @" . '".findresult'
-    silent exec '!comm -1 -3 <(sort "' . @" . '") <(sort "' . expand("%:p") . '") > "' . expand("%:p") . '".findresult'
-    silent exec '!cp -fv "' . @" . '.findresult' . '"' . ' ' . '"' . @" . '"'
-    silent exec '!cp -fv "' . expand("%:p") . '.findresult' . '"' . ' ' . '"' . expand("%:p") . '"'
+    exec '!~/loadrc/bashrc/cutcommon.sh ' . '"' .  @" . '"'  . ' ' . '"' .  expand("%:p") . '"'  
 endfunc
 
 nnoremap <leader>2 :call CutCommon()<cr>
