diff --git a/cscope b/cscope
index 55f601b2..59b12b1a 160000
--- ./cscope
+++ ./cscope
@@ -1 +1 @@
-Subproject commit 55f601b2519f45a5ee38b63d0d727f2f867eb33e
+Subproject commit 59b12b1a8ded4b02262376e44732f3594405eafd
diff --git a/neovim b/neovim
index 98bb5c89..3e715b8d 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit 98bb5c891d54f57de00205c730229bf8b8439894
+Subproject commit 3e715b8dfdfb9505be333a3060aa2499fe42ea00
