diff --git a/imvurc/ghypo.sh b/imvurc/ghypo.sh
index f04e3bfa..82761ec4 100755
--- ./imvurc/ghypo.sh
+++ ./imvurc/ghypo.sh
@@ -25,6 +25,6 @@ else
 
     if [ $(~/loadrc/gitrc/git_ready.sh) ]
     then
-        s/hypo
+        s/hypo --buildbot=hypo2 
     fi
 fi
diff --git a/neovim b/neovim
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit b0ed4793be6df3e1f951fd1874fbc5b3e266c91b
+Subproject commit b0ed4793be6df3e1f951fd1874fbc5b3e266c91b-dirty
