diff --git a/.shellrc b/.shellrc
index 52d99984..2c0a38b6 100755
--- ./.shellrc
+++ ./.shellrc
@@ -1,6 +1,5 @@
 #!/bin/zsh
 #alias pm-suspend='$HOME/loadrc/bashrc/sleep.sh'
-#alias sshfs='sshfs -o allow_other,uid=1000,gid=1000'
 #alias tmux='$HOME/loadrc/bashrc/tmux.sh'
 alias apf='$HOME/loadrc/bashrc/apf.sh'
 alias api='apt-get install -y'
@@ -74,7 +73,6 @@ alias sancho=/opt/sancho-0.9.4-59-linux-gtk/sancho
 alias selectCopy='$HOME/loadrc/bashrc/select_copy.sh'
 alias selectmirror='$HOME/loadrc/bashrc/selectmirror.sh'
 alias sleep='$HOME/loadrc/bashrc/sleep.sh'
-alias sload='$HOME/loadrc/bashrc/sload.sh'
 alias smount='$HOME/loadrc/bashrc/smount.sh'
 alias split='$HOME/loadrc/bashrc/split.sh'
 alias ssh='$HOME/loadrc/bashrc/ssh.sh'
diff --git a/bashrc/mount_samba.sh b/bashrc/mount_samba.sh
deleted file mode 100755
index fba69a67..00000000
--- ./bashrc/mount_samba.sh
+++ /dev/null
@@ -1,6 +0,0 @@
-#!/bin/zsh
-sshfs -o idmap=user huangyingw@ubuntu:/media/ /media/samba/
-if [ $? -ne 0 ]
-then
-    sshfs -o idmap=user huangyingw@vdonkey:/media/ /media/samba/
-fi
diff --git a/bashrc/sload.sh b/bashrc/sload.sh
deleted file mode 100755
index 1d01bc62..00000000
--- ./bashrc/sload.sh
+++ /dev/null
@@ -1,9 +0,0 @@
-#!/bin/zsh 
-if ping -c 1 macpro &> /dev/null
-then
-  sshfs -o nonempty huangyingw@macpro:~/Dropbox/loadrc/ ~/loadrc/
-  df -TH
-  . $HOME/loadrc/.loadrc
-else
-  . $HOME/loadrc/.loadrc
-fi
diff --git a/bashrc/smount.sh b/bashrc/smount.sh
index 8f22cb80..4cc616bb 100755
--- ./bashrc/smount.sh
+++ ./bashrc/smount.sh
@@ -10,11 +10,9 @@ mkdir -p ~/"$remoteServer"
 
 if [ $(uname) = "Linux" ]
 then
-    umount -l ~/"$remoteServer"
-    sshfs "$remoteServer":/ ~/"$remoteServer"
+    mount "$remoteServer":/media/volgrp/ ~/"$remoteServer"
     df -TH
 else
-    diskutil unmountDisk force ~/"$remoteServer"
-    sshfs "$remoteServer":/ ~/"$remoteServer"
+    sudo mount -o nolocks -o resvport "$remoteServer":/media/volgrp/ ~/"$remoteServer"
     df -H
 fi
diff --git a/gitrc/git_diff_command.sh b/gitrc/git_diff_command.sh
index e2e8aa4b..704a0384 100755
--- ./gitrc/git_diff_command.sh
+++ ./gitrc/git_diff_command.sh
@@ -1,3 +1,8 @@
 #!/bin/zsh
 
-echo "git diff --ignore-space-at-eol -b -w --ignore-blank-lines --binary"
+echo git diff \
+    --ignore-space-at-eol \
+    --ignore-space-change \
+    --ignore-all-space \
+    --ignore-blank-lines \
+    --binary
diff --git a/install_basic_macos.sh b/install_basic_macos.sh
index 270f3cb2..fb1e2f26 100755
--- ./install_basic_macos.sh
+++ ./install_basic_macos.sh
@@ -37,7 +37,6 @@ brew install \
     mosh \
     openssl \
     rename \
-    sshfs \
     tig \
     tmux \
     w3m \
diff --git a/vimrc/.vim/plugin/view.vim b/vimrc/.vim/plugin/view.vim
index 41afb6c6..56af3e53 100644
--- ./vimrc/.vim/plugin/view.vim
+++ ./vimrc/.vim/plugin/view.vim
@@ -38,6 +38,7 @@ else
 endif
 
 set diffopt+=iwhiteall
+set diffopt+=iblank
 set diffopt+=algorithm:patience
 set diffopt+=indent-heuristic
 set wrap
