diff --git a/bashrc/jdupes.sh b/bashrc/jdupes.sh
index ccc72429..0527d6b9 100755
--- ./bashrc/jdupes.sh
+++ ./bashrc/jdupes.sh
@@ -22,5 +22,3 @@ then \
         ~/loadrc/bashrc/sleep.sh ; \
         fi ; \
         nvim ~/loadrc/jdupes.log
-
-echo check > /sys/block/md0/md/sync_action
diff --git a/vimrc/.vim/bundle/ncm2 b/vimrc/.vim/bundle/ncm2
index 7d49f7b4..3de17695 160000
--- ./vimrc/.vim/bundle/ncm2
+++ ./vimrc/.vim/bundle/ncm2
@@ -1 +1 @@
-Subproject commit 7d49f7b473db732878091f2f8c68bc47b015559b
+Subproject commit 3de17695c58f309642c5148b21d47fac61937a96
diff --git a/vimrc/.vim/plugin/view.vim b/vimrc/.vim/plugin/view.vim
index 41afb6c6..10c8eeb3 100644
--- ./vimrc/.vim/plugin/view.vim
+++ ./vimrc/.vim/plugin/view.vim
@@ -37,6 +37,7 @@ else
     set winwidth=999999
 endif
 
+set diffopt+=iblank
 set diffopt+=iwhiteall
 set diffopt+=algorithm:patience
 set diffopt+=indent-heuristic
