diff --git a/bashrc/fvideos.sh b/bashrc/fvideos.sh
index 03d4d427..38fb060b 100755
--- ./bashrc/fvideos.sh
+++ ./bashrc/fvideos.sh
@@ -2,6 +2,7 @@
 
 FAVLOG="fav.log.bak"
 FAVLOGSORT="fav.log.sort.bak"
+export LC_ALL=C
 find . -type f -size +500k -exec du -h {} + | sort -r -h | cut -f 2 | sed 's/\(["\]\)/\\\1/g;s/.*/"&"/' > "$FAVLOG" && \
     cp -fv "$FAVLOG" fav.log && \
     find . -type f -size +500k -exec ls -t {} \+ | sed 's/\(["\]\)/\\\1/g;s/.*/"&"/' > "$FAVLOGSORT" && \
diff --git a/bashrc/jdupes.sh b/bashrc/jdupes.sh
index 0527d6b9..b348c7ed 100755
--- ./bashrc/jdupes.sh
+++ ./bashrc/jdupes.sh
@@ -16,7 +16,7 @@ if [ -f ${MIRRORCHECK} ] ; \
 then \
     ~/loadrc/bashrc/move67.sh ; \
     fi && \
-    jdupes -1dNr -X size-:100k -X nostr:/.git/ "$TARGET" 2>&1 | tee ~/loadrc/jdupes.log && \
+    jdupes -1dNr -X size-:100m -X nostr:/.git/ "$TARGET" 2>&1 | tee ~/loadrc/jdupes.log && \
     if [ -f ${MIRRORCHECK} ] ; \
     then \
         ~/loadrc/bashrc/sleep.sh ; \
diff --git a/error b/error
index 2c5a88c9..c0e586ef 100644
--- ./error
+++ ./error
@@ -62,3 +62,5 @@ Cought anything: Vim(cd):E472: Command failed
 
 [ncm2_go@yarp] TypeError: expected str, bytes or os.PathLike object, not NoneType
 [ncm2_pyclang_proc@yarp] try_notify cache_add failed, job is dead
+
+The legacy SnipMate parser is deprecated. Please see :h SnipMate-deprecate.
diff --git a/install_prerequisite.sh b/install_prerequisite.sh
index 20e3534e..298b6bc4 100755
--- ./install_prerequisite.sh
+++ ./install_prerequisite.sh
@@ -27,6 +27,8 @@ apt-get install -y \
     hfsprogs \
     htop \
     inetutils-traceroute \
+    iperf \
+    iperf3 \
     iputils-ping \
     kdiff3 \
     libsecret-1-dev \
diff --git a/vimrc/.vim/bundle/Dockerfile.vim b/vimrc/.vim/bundle/Dockerfile.vim
index 03848056..2386923a 160000
--- ./vimrc/.vim/bundle/Dockerfile.vim
+++ ./vimrc/.vim/bundle/Dockerfile.vim
@@ -1 +1 @@
-Subproject commit 038480563ce04c5b5e233c5ff6044db2a0760625
+Subproject commit 2386923a8c30d0c4f06f27eef2a025752423ea86
