diff --git a/bashrc/fvideos.sh b/bashrc/fvideos.sh
index 03d4d427..38fb060b 100755
--- ./bashrc/fvideos.sh
+++ ./bashrc/fvideos.sh
@@ -2,6 +2,7 @@
 
 FAVLOG="fav.log.bak"
 FAVLOGSORT="fav.log.sort.bak"
+export LC_ALL=C
 find . -type f -size +500k -exec du -h {} + | sort -r -h | cut -f 2 | sed 's/\(["\]\)/\\\1/g;s/.*/"&"/' > "$FAVLOG" && \
     cp -fv "$FAVLOG" fav.log && \
     find . -type f -size +500k -exec ls -t {} \+ | sed 's/\(["\]\)/\\\1/g;s/.*/"&"/' > "$FAVLOGSORT" && \
diff --git a/error b/error
index 2c5a88c9..c0e586ef 100644
--- ./error
+++ ./error
@@ -62,3 +62,5 @@ Cought anything: Vim(cd):E472: Command failed
 
 [ncm2_go@yarp] TypeError: expected str, bytes or os.PathLike object, not NoneType
 [ncm2_pyclang_proc@yarp] try_notify cache_add failed, job is dead
+
+The legacy SnipMate parser is deprecated. Please see :h SnipMate-deprecate.
diff --git a/install_basic_macos.sh b/install_basic_macos.sh
index 260d1c88..ba79a071 100755
--- ./install_basic_macos.sh
+++ ./install_basic_macos.sh
@@ -22,6 +22,8 @@ brew install \
     hg \
     htop \
     hub \
+    iperf \
+    iperf3 \
     jupyterlab \
     kubectl \
     kubernetes-cli \
diff --git a/install_prerequisite.sh b/install_prerequisite.sh
index 20e3534e..298b6bc4 100755
--- ./install_prerequisite.sh
+++ ./install_prerequisite.sh
@@ -27,6 +27,8 @@ apt-get install -y \
     hfsprogs \
     htop \
     inetutils-traceroute \
+    iperf \
+    iperf3 \
     iputils-ping \
     kdiff3 \
     libsecret-1-dev \
