diff --git a/.gitrc b/.gitrc
index 31ff4791..8e753cb1 100755
--- ./.gitrc
+++ ./.gitrc
@@ -11,7 +11,6 @@ alias gbr='$HOME/loadrc/gitrc/gbr.sh'
 alias gbra='$HOME/loadrc/gitrc/gbra.sh'
 alias gbrd='$HOME/loadrc/gitrc/gbrd.sh'
 alias gbrm='$HOME/loadrc/gitrc/gbrm.sh'
-alias gbrs='$HOME/loadrc/gitrc/gbrs.sh'
 alias gch='git cherry'
 alias gchp='git cherry-pick'
 alias gcia='$HOME/loadrc/gitrc/gcia.sh'
diff --git a/gitrc/dogdio.sh b/gitrc/dogdio.sh
index 4f8edfc6..c32f5329 100755
--- ./gitrc/dogdio.sh
+++ ./gitrc/dogdio.sh
@@ -12,4 +12,5 @@ remote="$(git config gsync.remote)"
 
 ~/loadrc/gitrc/gdi.sh "$remote/$remote_branch" 2>&1 | tee "$output"  
 sed -i.bak "s/^--- a\//--- \.\//g;s/^+++ b\//+++ \.\//g" "$output"
+dos2unix "$output" 
 exit 0
diff --git a/gitrc/gdit.sh b/gitrc/gdit.sh
index 700d6af6..8e39a3ec 100755
--- ./gitrc/gdit.sh
+++ ./gitrc/gdit.sh
@@ -15,6 +15,7 @@ git merge -X theirs "$remote/$remote_branch" && \
     git push ; \
     git checkout "$current_branch" && \
     ~/loadrc/gitrc/gdi.sh "$target_branch " "$current_branch" 2>&1 | tee "$output"
+    dos2unix "$output" 
 
 if [ $? -ne 0 ]
 then
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index ffc1c04f..f2f44f65 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit ffc1c04f455c2ecd203ebac0731d4e9a214cbcaa
+Subproject commit f2f44f65ada0a750b06951ff53ece163e39575ac
