diff --git a/.gitconfig_sample b/.gitconfig_sample
index 639febb9..0ea38ff6 100644
--- ./.gitconfig_sample
+++ ./.gitconfig_sample
@@ -8,7 +8,7 @@
     remote = origin
 [gsync]
     remote = authorVar
-    target = authorVar/master
+    target = authorVar/branchVar
 [merge]
     allow = true
 [branch "dev"]
diff --git a/gitrc/gcl.sh b/gitrc/gcl.sh
index e118a4de..e1f58816 100755
--- ./gitrc/gcl.sh
+++ ./gitrc/gcl.sh
@@ -18,7 +18,10 @@ urlVar=$(echo "$urlVar" | sed 's/\//\\\//g')
 sed -i.bak "s|urlVar|$urlVar|g;s|repoVar|${author}_${repo}|g;s|authorVar|$author|g" "$target"/.gitconfig
 echo ".gitconfig --> $(realpath ${target}/.gitconfig)"
 
-cd "$target" && \
+cd "$target"
+CURRENT_BRANCH=$(~/loadrc/gitrc/get_current_branch.sh)
+sed -i.bak "s|branchVar|$CURRENT_BRANCH|g" .gitconfig
+
 git remote remove origin ; \
     $HOME/loadrc/gitrc/gcob.sh dev && \
     $HOME/loadrc/gitrc/gdev.sh && \
diff --git a/neovim b/neovim
index 03e622e5..277694e5 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit 03e622e58686d209978eddf625a5f1ab7b69d491
+Subproject commit 277694e5e1777ff9ce35fc914cfaccdc27e26236
diff --git a/vimrc/.vim/bundle/neco-vim b/vimrc/.vim/bundle/neco-vim
index ec50b3cb..ba9b6535 160000
--- ./vimrc/.vim/bundle/neco-vim
+++ ./vimrc/.vim/bundle/neco-vim
@@ -1 +1 @@
-Subproject commit ec50b3cbb4537625c2aaba7601232a53ccd779aa
+Subproject commit ba9b6535381690fc6773d682fc046d8ddd2a863a
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 7b0b8fb5..a8d0cbeb 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 7b0b8fb54c39cf0027c356832b5b55283746eef6
+Subproject commit a8d0cbebf04cfe557abefbd0604a0613f518052c
