diff --git a/.gitrc b/.gitrc
index 621cf604..f4e4d1cd 100755
--- ./.gitrc
+++ ./.gitrc
@@ -22,7 +22,6 @@ alias gclean='git clean -fd'
 alias gco='$HOME/loadrc/gitrc/gco.sh'
 alias gcob='$HOME/loadrc/gitrc/gcob.sh'
 alias gcof='$HOME/loadrc/gitrc/gcof.sh'
-alias gcom='$HOME/loadrc/gitrc/gcom.sh'
 alias gctb='$HOME/loadrc/gitrc/gctb.sh'
 alias gdev='$HOME/loadrc/gitrc/gdev.sh'
 alias gdi2='$HOME/loadrc/gitrc/gdi2.sh'
diff --git a/gitrc/discard_unnecessaries.sh b/gitrc/discard_unnecessaries.sh
new file mode 100755
index 00000000..7e8bfedd
--- /dev/null
+++ ./gitrc/discard_unnecessaries.sh
@@ -0,0 +1,5 @@
+#!/bin/zsh
+git checkout \
+    *.gdio.diff \
+    files.proj \
+    rsync.files
diff --git a/gitrc/gbib.sh b/gitrc/gbib.sh
index 36163b51..f0c27725 100755
--- ./gitrc/gbib.sh
+++ ./gitrc/gbib.sh
@@ -1,3 +1,4 @@
 #!/bin/zsh 
-git checkout files.proj ; git bisect bad
+~/loadrc/gitrc/discard_unnecessaries.sh
+git bisect bad
 git bisect log | tee gbil.log
diff --git a/gitrc/gbig.sh b/gitrc/gbig.sh
index e81f6b28..067a5b7b 100755
--- ./gitrc/gbig.sh
+++ ./gitrc/gbig.sh
@@ -1,3 +1,4 @@
 #!/bin/zsh
-git checkout files.proj ; git bisect good
+~/loadrc/gitrc/discard_unnecessaries.sh
+git bisect good
 git bisect log | tee gbil.log
diff --git a/gitrc/gcob.sh b/gitrc/gcob.sh
index 7a69b85f..ae272238 100755
--- ./gitrc/gcob.sh
+++ ./gitrc/gcob.sh
@@ -6,6 +6,7 @@ then
 fi
 
 branch="$1"
+currentBranch=$(~/loadrc/gitrc/get_current_branch.sh)
 git remote update
 
 if [ -n "$2" ]
@@ -21,3 +22,7 @@ else
         git checkout -b "$branch"
     fi
 fi
+
+~/loadrc/gitrc/discard_unnecessaries.sh
+git checkout "$branch"
+git merge "$currentBranch"
diff --git a/gitrc/gcom.sh b/gitrc/gcom.sh
deleted file mode 100755
index 2312b524..00000000
--- ./gitrc/gcom.sh
+++ /dev/null
@@ -1,3 +0,0 @@
-#!/bin/zsh
-~/loadrc/gitrc/gme2.sh "$1" && \
-    git checkout "$1"
diff --git a/gitrc/gdi2.sh b/gitrc/gdi2.sh
index 53c8e7dc..1fd66786 100755
--- ./gitrc/gdi2.sh
+++ ./gitrc/gdi2.sh
@@ -28,7 +28,7 @@ local_master="$(git config gsync.branch)"
 
 GDITDIFF=$(echo "$currentBranch.gdit.diff" | sed 's/\//_/g')
 git checkout -b "$targetBranch" "$local_master" ; \
-    git checkout files.proj ; \
+    ~/loadrc/gitrc/discard_unnecessaries.sh ; \
     git checkout "$targetBranch" ; \
     git apply --reject --whitespace=fix "$GDITDIFF" ; \
     ~/loadrc/gitrc/checkout_rejs.sh "$currentBranch" && \
diff --git a/gitrc/gdit.sh b/gitrc/gdit.sh
index 72eac338..4958da0c 100755
--- ./gitrc/gdit.sh
+++ ./gitrc/gdit.sh
@@ -5,7 +5,7 @@ target_branch=$(echo "$current_branch" | sed 's/\.fix$//g')
 remote="$(git config gsync.remote)"
 remote_branch="$(git config gsync.branch)"
 
-git checkout files.proj
+~/loadrc/gitrc/discard_unnecessaries.sh
 git checkout -b "$target_branch" "$remote/$target_branch"
 git checkout -b "$target_branch" "$remote/$remote_branch"
 git checkout -f "$target_branch"
diff --git a/gitrc/gpl.sh b/gitrc/gpl.sh
index 9b5215c0..db0026a7 100755
--- ./gitrc/gpl.sh
+++ ./gitrc/gpl.sh
@@ -1,7 +1,7 @@
 #!/bin/zsh
 ~/loadrc/gitrc/include_gitconfig.sh
 git remote update
-git checkout files.proj *.gdio.diff
+~/loadrc/gitrc/discard_unnecessaries.sh
 git pull
 currentBranch=$(~/loadrc/gitrc/get_current_branch.sh)
 
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index c3e993d7..bcb2b652 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -26,7 +26,6 @@ command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gclean
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gco :execute s:Gco(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gcob :execute s:Gcob(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gcof :execute s:Gcof(<f-args>)
-command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gcom :execute s:Gcom(<q-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gcp :execute s:Gcp(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gdev :execute s:Gdev()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gdi :execute s:Gdi(<f-args>)
@@ -496,12 +495,6 @@ function! s:KdiffFile() abort
     call asyncrun#run('<bang>', '', '~/loadrc/leetcoderc/KdiffFile.py ' . '"' .  expand("%:p") . '"')
 endfunction
 
-
-function! s:Gcom(args, ...) abort
-    let worktree = Cd2Worktree()
-    call asyncrun#run('<bang>', '', '~/loadrc/gitrc/gcom.sh ' . '"' .  a:args . '"')
-endfunction
-
 function! s:Gshow(args, ...) abort
     let worktree = Cd2Worktree()
     let output = a:args . '.diff'
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 8ffb999b..5501ef26 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -293,7 +293,7 @@ function! VimOpen()
     elseif (expand("%") ==# 'gbil.log')
         let b:commit = expand("<cword>")
         call Cd2Worktree()
-        exec '!git checkout files.proj ; git checkout ' . '"' .  b:commit . '"'
+        exec '!~/loadrc/gitrc/discard_unnecessaries.sh ; git checkout ' . '"' .  b:commit . '"'
     elseif (expand("%") ==# 'glg.findresult')
         let b:commit = expand("<cword>")
         exec '!git checkout ' . '"' .  b:commit . '"'
@@ -305,7 +305,7 @@ function! VimOpen()
     elseif (&filetype ==# 'fugitiveblame')
         let b:commit = expand("<cword>")
         call Cd2Worktree()
-        exec '!git checkout files.proj ; git diff --quiet && git diff HEAD --quiet &&  git checkout -f ' . '"' .  b:commit . '^"'
+        exec '!~/loadrc/gitrc/discard_unnecessaries.sh ; git diff --quiet && git diff HEAD --quiet &&  git checkout -f ' . '"' .  b:commit . '^"'
         ""call fugitive#Command(<line1>, <count>, +"<range>", <bang>0, "<mods>", "' . 'blame' . ' " . <q-args>)
         "let s:cmd = 'Blame'
         "exec 'command! -bang -nargs=? -range=-1 -complete=customlist,fugitive#' . s:cmd . 'Complete G' . tolower(s:cmd) 'exe fugitive#Command(<line1>, <count>, +"<range>", <bang>0, "<mods>", "' . tolower(s:cmd) . ' " . <q-args>)'
diff --git a/vishrc/vrun.sh b/vishrc/vrun.sh
index 555037d8..99707d6d 100755
--- ./vishrc/vrun.sh
+++ ./vishrc/vrun.sh
@@ -3,7 +3,7 @@ file="$1"
 
 if [ $(basename "$file") = "gbil.log" ]
 then
-    git checkout files.proj
+    ~/loadrc/gitrc/discard_unnecessaries.sh
     ~/loadrc/gitrc/gbir.sh
     exit 0
 fi
