diff --git a/gitrc/gdi2.sh b/gitrc/gdi2.sh
index fb80774f..dbcf937b 100755
--- ./gitrc/gdi2.sh
+++ ./gitrc/gdi2.sh
@@ -28,9 +28,8 @@ rpath=$(git config deploy.path)
 GDITDIFF=$(echo "$currentBranch.gdit.diff" | sed 's/\//_/g')
     ~/loadrc/gitrc/discard_unnecessaries.sh ; \
     git checkout -f "$targetBranch" ; \
-    git apply --reject --whitespace=fix "$GDITDIFF" ; \
+    git apply --index --reject --whitespace=fix "$GDITDIFF" ; \
     ~/loadrc/gitrc/checkout_rejs.sh "$currentBranch" && \
-    git add . && \
     git commit  --no-verify -am "$commit_message" ; \
     git pull ; \
     git push ; \
diff --git a/gitrc/gdif.sh b/gitrc/gdif.sh
index 10543369..0865ae41 100755
--- ./gitrc/gdif.sh
+++ ./gitrc/gdif.sh
@@ -1,6 +1,6 @@
 #!/bin/zsh
 
-COMMAND="git diff --no-prefix --ignore-space-at-eol -b -w --ignore-blank-lines HEAD $1"
+COMMAND="git diff --ignore-space-at-eol -b -w --ignore-blank-lines HEAD $1"
 
 while read ss
 do
diff --git a/vimrc/.vim/bundle/vim-anyfold b/vimrc/.vim/bundle/vim-anyfold
index 4579327e..662a25ea 160000
--- ./vimrc/.vim/bundle/vim-anyfold
+++ ./vimrc/.vim/bundle/vim-anyfold
@@ -1 +1 @@
-Subproject commit 4579327e2f73dbf1018d58df9bd356d140059d2b
+Subproject commit 662a25ea7f07ab2bcd816a54770dc724149e4e1a
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index 8d23e01b..eb53a833 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit 8d23e01bc6a21f40acbd1ea6d357ae3490d4c607
+Subproject commit eb53a83353295179cf7774ca57c7cac9651f0afc
diff --git a/vimrc/.vim/bundle/vim-buffergator b/vimrc/.vim/bundle/vim-buffergator
index 8d2ff564..7178f722 160000
--- ./vimrc/.vim/bundle/vim-buffergator
+++ ./vimrc/.vim/bundle/vim-buffergator
@@ -1 +1 @@
-Subproject commit 8d2ff5644ea3bcfccc36b75d66c6ffd287922b1f
+Subproject commit 7178f7221b6923ec3063820c7dc583a25fc60fda
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 2954fe26..edcf19d8 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 2954fe260004b8b2734daee63a43d2d6bb725884
+Subproject commit edcf19d82f3c46d15394d08d1b434135ddcb8596
diff --git a/vimrc/.vim/bundle/vim-javascript-syntax b/vimrc/.vim/bundle/vim-javascript-syntax
index d64ef4d1..b814218e 160000
--- ./vimrc/.vim/bundle/vim-javascript-syntax
+++ ./vimrc/.vim/bundle/vim-javascript-syntax
@@ -1 +1 @@
-Subproject commit d64ef4d1f9dbb1f141feab28d58ab6ee0e4ac769
+Subproject commit b814218eb0c31a7f49cf7b1121b32d11a6f4543d
diff --git a/vimrc/.vim/bundle/vim-log-syntax b/vimrc/.vim/bundle/vim-log-syntax
--- ./vimrc/.vim/bundle/vim-log-syntax
+++ ./vimrc/.vim/bundle/vim-log-syntax
@@ -1 +1 @@
-Subproject commit e2dcef439e50fec752177fc1aa46b57114e31698
+Subproject commit e2dcef439e50fec752177fc1aa46b57114e31698-dirty
diff --git a/vimrc/.vim/bundle/vim-snipmate b/vimrc/.vim/bundle/vim-snipmate
index 68eebf9d..b80367fe 160000
--- ./vimrc/.vim/bundle/vim-snipmate
+++ ./vimrc/.vim/bundle/vim-snipmate
@@ -1 +1 @@
-Subproject commit 68eebf9dabe5c1d1964e78dda188765f224bab34
+Subproject commit b80367fefc2188c9d6c4a9eb1830360b09e13476
