diff --git a/bashrc/update_proj.sh b/bashrc/update_proj.sh
index b816b9d8..6bf1ff3a 100755
--- ./bashrc/update_proj.sh
+++ ./bashrc/update_proj.sh
@@ -1,3 +1,3 @@
 #!/bin/zsh
-. ~/loadrc/bashrc/find_up_goto.sh "files.proj"
+~/loadrc/bashrc/find_up_goto.sh "files.proj"
 ~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/cscope.sh
diff --git a/gitrc/gdi.sh b/gitrc/gdi.sh
index 4799a9c8..b720ac69 100755
--- ./gitrc/gdi.sh
+++ ./gitrc/gdi.sh
@@ -1,5 +1,6 @@
 #!/bin/zsh
-COMMAND="git diff --ignore-space-at-eol -b -w --ignore-blank-lines"
+
+COMMAND=$(~/loadrc/gitrc/git_diff_command.sh)
 
 if [ -z "$1" ]
 then
diff --git a/gitrc/gdif.sh b/gitrc/gdif.sh
index 0865ae41..62a3d1c6 100755
--- ./gitrc/gdif.sh
+++ ./gitrc/gdif.sh
@@ -1,6 +1,7 @@
 #!/bin/zsh
 
-COMMAND="git diff --ignore-space-at-eol -b -w --ignore-blank-lines HEAD $1"
+COMMAND=$(~/loadrc/gitrc/git_diff_command.sh)
+COMMAND="$COMMAND HEAD $1"
 
 while read ss
 do
diff --git a/gitrc/git_diff_command.sh b/gitrc/git_diff_command.sh
new file mode 100755
index 00000000..601480d6
--- /dev/null
+++ ./gitrc/git_diff_command.sh
@@ -0,0 +1,3 @@
+#!/bin/zsh
+
+echo "git diff --ignore-space-at-eol -b -w --ignore-blank-lines"
diff --git a/vimrc/.vim/bundle/indentLine b/vimrc/.vim/bundle/indentLine
index 9662ef8f..a03953d4 160000
--- ./vimrc/.vim/bundle/indentLine
+++ ./vimrc/.vim/bundle/indentLine
@@ -1 +1 @@
-Subproject commit 9662ef8f0443211b11fd87919343d12179bca548
+Subproject commit a03953d4e89ebc372674f88303c5d4933709fea6
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index bb9b8394..4fa2506d 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit bb9b8394df0d23c136aee1305f6ee4bf0e20b709
+Subproject commit 4fa2506dadc2f667ba84d739151961f53678f529
