diff --git a/.gitrc b/.gitrc
index af69691c..621cf604 100755
--- ./.gitrc
+++ ./.gitrc
@@ -28,6 +28,8 @@ alias gdev='$HOME/loadrc/gitrc/gdev.sh'
 alias gdi2='$HOME/loadrc/gitrc/gdi2.sh'
 alias gdi='$HOME/loadrc/gitrc/gdi.sh'
 alias gdif='$HOME/loadrc/gitrc/gdif.sh'
+alias gdio='$HOME/loadrc/gitrc/gdio.sh'
+alias gdit='$HOME/loadrc/gitrc/gdit.sh'
 alias gfe='git fetch'
 alias gff='$HOME/loadrc/gitrc/gff.sh'
 alias gfix='$HOME/loadrc/gitrc/gfix.sh'
diff --git a/bashrc/cscope.sh b/bashrc/cscope.sh
index e9c3e6c6..ba99cae7 100755
--- ./bashrc/cscope.sh
+++ ./bashrc/cscope.sh
@@ -56,7 +56,6 @@ find . "(" "${prune_params[@]}" ")" -a -prune -o -size +0 -type f -exec grep -Il
     sed -i.bak 's/ /\\ /g' files.proj && \
     ~/loadrc/bashrc/fvideos.sh && \
     ~/loadrc/bashrc/fdocs.sh && \
-    echo > cscope.small.files && \
     echo "$TARGETEDIR"/files.proj | sed 's/\(["'\''\]\)/\\\1/g;s/ /\\ /g;s/.*/"&"/' >> ~/all.proj && \
     cscope -bq -i "$TARGET" -f cscope.out.bak && \
     cp -fv cscope.out.bak cscope.out && \
diff --git a/gitrc/dogdio.sh b/gitrc/dogdio.sh
index 6d71ac21..fb26b258 100755
--- ./gitrc/dogdio.sh
+++ ./gitrc/dogdio.sh
@@ -4,8 +4,6 @@ then
     exit 0
 fi
 
-~/loadrc/gitrc/gpl.sh
-
 local_branch="$(~/loadrc/gitrc/get_current_branch.sh)" 
 output="$local_branch.gdio.diff"
 output=$(echo "$output" | sed 's/\//_/g')
diff --git a/gitrc/dogsync.sh b/gitrc/dogsync.sh
index aee86c9d..76ff478a 100755
--- ./gitrc/dogsync.sh
+++ ./gitrc/dogsync.sh
@@ -1,3 +1,4 @@
 #!/bin/zsh
 ~/loadrc/gitrc/gpl.sh &
 ~/loadrc/gitrc/gps.sh &
+~/loadrc/gitrc/dogdio.sh &
diff --git a/gitrc/gdit.sh b/gitrc/gdit.sh
index f9684f2e..6de01696 100755
--- ./gitrc/gdit.sh
+++ ./gitrc/gdit.sh
@@ -7,7 +7,8 @@ remote_branch="$(git config gsync.branch)"
 
 git checkout files.proj
 git checkout -b "$target_branch" "$remote_branch"
-git merge "$remote/$remote_branch" && \
+git checkout -f "$target_branch"
+git merge -X theirs "$remote/$remote_branch" && \
     git pull && \
     git push ; \
     git checkout "$current_branch" && \
diff --git a/gitrc/gicb.sh b/gitrc/gicb.sh
index bec88869..bd3c7b59 100755
--- ./gitrc/gicb.sh
+++ ./gitrc/gicb.sh
@@ -7,4 +7,5 @@ git init \
     && git add . \
     && git add -f files.proj \
     && $HOME/loadrc/gitrc/g.sh \
+    && hub create \
     && ~/loadrc/gitrc/gclb.sh
diff --git a/gitrc/gpl.sh b/gitrc/gpl.sh
index 0ec0480c..5aa5d719 100755
--- ./gitrc/gpl.sh
+++ ./gitrc/gpl.sh
@@ -1,7 +1,7 @@
 #!/bin/zsh
 ~/loadrc/gitrc/include_gitconfig.sh
 git remote update
-git checkout files.proj
+git checkout files.proj *.gdio.diff
 git pull
 currentBranch=$(~/loadrc/gitrc/get_current_branch.sh)
 
diff --git a/imvurc/ghypo.sh b/imvurc/ghypo.sh
index f72f4122..e8af707a 100755
--- ./imvurc/ghypo.sh
+++ ./imvurc/ghypo.sh
@@ -8,8 +8,9 @@ else
     ~/loadrc/gitrc/gclean.sh
     git remote update
     git checkout -b "$1" origin/"$1"
-    git checkout "$1"
-    git pull
+    git checkout -f "$1"
+    git branch -u origin/"$1"
+    git merge -X theirs origin/"$1" 
 
     if [ $(~/loadrc/gitrc/git_ready.sh) ]
     then
diff --git a/prunefile_template.conf b/prunefile_template.conf
index fb57bfdc..2154bcb9 100644
--- ./prunefile_template.conf
+++ ./prunefile_template.conf
@@ -1,5 +1,4 @@
 "./cscope"
-"./cscope.small.files"
 "./files.proj"
 "./running.lck"
 "./running.time"
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 895f4400..c3e993d7 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -347,6 +347,10 @@ function! s:Gdi(...) abort
     if expand('%:t') != 'index'
         if arg1 == ''
             call fugitive#Diffsplit(0, 1, "vert", '', [])
+        elseif tolower(arg1) == 'o'
+            let remote = substitute(system("git config gsync.remote"), '\n', '', '')
+            let branch = substitute(system("git config gsync.branch"), '\n', '', '')
+            call fugitive#Diffsplit(1, 0, '', remote . '/' . branch, [remote . '/' . branch])
         else
             call fugitive#Diffsplit(0, 1, "vert", arg1, [arg1])
         endif
@@ -382,17 +386,9 @@ function! s:Gdit() abort
 endfunction
 
 function! s:Gdio(...) abort
-    let remote = substitute(system("git config gsync.remote"), '\n', '', '')
-    let branch = substitute(system("git config gsync.branch"), '\n', '', '')
-    if expand('%:t') != 'index'
-        call fugitive#Diffsplit(1, 0, '', remote . '/' . branch, [remote . '/' . branch])
-        return 
-    endif
-
     let worktree = Cd2Worktree()
     let current_branch = substitute(system("~/loadrc/gitrc/get_current_branch.sh"), '\n', '', '')
-    let local_branch = (a:0 >= 1) ? a:1 : current_branch
-    let output = local_branch . '.gdio.diff'
+    let output = current_branch . '.gdio.diff'
     let output = substitute(output, "/", "_", "g")
     exec '!~/loadrc/gitrc/gdio.sh'
 
