diff --git a/gitrc/dogdio.sh b/gitrc/dogdio.sh
index 6d71ac21..fb26b258 100755
--- ./gitrc/dogdio.sh
+++ ./gitrc/dogdio.sh
@@ -4,8 +4,6 @@ then
     exit 0
 fi
 
-~/loadrc/gitrc/gpl.sh
-
 local_branch="$(~/loadrc/gitrc/get_current_branch.sh)" 
 output="$local_branch.gdio.diff"
 output=$(echo "$output" | sed 's/\//_/g')
diff --git a/gitrc/dogsync.sh b/gitrc/dogsync.sh
index aee86c9d..76ff478a 100755
--- ./gitrc/dogsync.sh
+++ ./gitrc/dogsync.sh
@@ -1,3 +1,4 @@
 #!/bin/zsh
 ~/loadrc/gitrc/gpl.sh &
 ~/loadrc/gitrc/gps.sh &
+~/loadrc/gitrc/dogdio.sh &
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 895f4400..71798371 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -384,24 +384,7 @@ endfunction
 function! s:Gdio(...) abort
     let remote = substitute(system("git config gsync.remote"), '\n', '', '')
     let branch = substitute(system("git config gsync.branch"), '\n', '', '')
-    if expand('%:t') != 'index'
     call fugitive#Diffsplit(1, 0, '', remote . '/' . branch, [remote . '/' . branch])
-        return 
-    endif
-
-    let worktree = Cd2Worktree()
-    let current_branch = substitute(system("~/loadrc/gitrc/get_current_branch.sh"), '\n', '', '')
-    let local_branch = (a:0 >= 1) ? a:1 : current_branch
-    let output = local_branch . '.gdio.diff'
-    let output = substitute(output, "/", "_", "g")
-    exec '!~/loadrc/gitrc/gdio.sh'
-
-    if bufwinnr('^' . output . '$') > 0
-        exe "bd!" . output
-    endif
-
-    let worktree = Cd2Worktree()
-    call OpenOrSwitch(output, 'vs')
 endfunction
 
 function! s:Gdi2(...) abort
