diff --git a/bashrc/select_move.sh b/bashrc/select_move.sh
new file mode 100755
index 00000000..29fd5336
--- /dev/null
+++ ./bashrc/select_move.sh
@@ -0,0 +1,8 @@
+#!/bin/zsh
+
+cat "$1" | sed 's/^"//g;s/"$//g;s/\\ / /g' > "$1".tmp
+rsync -aHv --force --progress \
+    --remove-source-files \
+    --files-from "$1".tmp \
+    "$2" \
+    "$3"
diff --git a/gitrc/gcom.sh b/gitrc/gcom.sh
index 83a83235..e8b5bfa5 100755
--- ./gitrc/gcom.sh
+++ ./gitrc/gcom.sh
@@ -25,3 +25,4 @@ then
 fi
 
 ~/loadrc/gitrc/gdio.sh
+git checkout "$CURRENT_BRANCH.gdio.diff"
diff --git a/gitrc/gdi2.sh b/gitrc/gdi2.sh
index 2e325ba5..88dedbbc 100755
--- ./gitrc/gdi2.sh
+++ ./gitrc/gdi2.sh
@@ -25,6 +25,9 @@ fi
 
 host=$(git config deploy.host)
 rpath=$(git config deploy.path)
+remote="$(git config gsync.remote)"
+
+git push "$remote" "$targetBranch"
 
 GDITDIFF=$(echo "$currentBranch.gdit.diff" | sed 's/\//_/g')
 
diff --git a/gitrc/gdit.sh b/gitrc/gdit.sh
index ad9a7f1a..02d50a52 100755
--- ./gitrc/gdit.sh
+++ ./gitrc/gdit.sh
@@ -11,10 +11,11 @@ git checkout -f "$target_branch"
 git branch -u "$remote/$target_branch"
 git merge -X theirs "$target" && \
     git pull && \
-    git push ; \
     git checkout "$current_branch" && \
     ~/loadrc/gitrc/gdi.sh "$target_branch " "$current_branch" 2>&1 | tee "$output"
 
+git fetch
+
 if [ $? -ne 0 ]
 then
     echo -e "${red}most probably sth wrong in git merge... ${NC}" 2>&1 | tee "$output"
diff --git a/imvurc/ghypo.sh b/imvurc/ghypo.sh
index f04e3bfa..bd703fed 100755
--- ./imvurc/ghypo.sh
+++ ./imvurc/ghypo.sh
@@ -26,5 +26,6 @@ else
     if [ $(~/loadrc/gitrc/git_ready.sh) ]
     then
         s/hypo
+        #s/hypo --buildbot=hypo2 
     fi
 fi
diff --git a/neovim b/neovim
index 657531e3..b5c09465 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit 657531e380d492eb2f1e011cb63a1cbfebd71f18
+Subproject commit b5c09465eb83da5cd1cb33ed6cbc9b4b5fad2199
diff --git a/vimrc/.vim/bundle/ncm2 b/vimrc/.vim/bundle/ncm2
index 3de17695..d2d8cc43 160000
--- ./vimrc/.vim/bundle/ncm2
+++ ./vimrc/.vim/bundle/ncm2
@@ -1 +1 @@
-Subproject commit 3de17695c58f309642c5148b21d47fac61937a96
+Subproject commit d2d8cc430232f385d8468f44da8635febb9a92ad
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index 1bb095f1..fb49cea4 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit 1bb095f1cd592b5e381f89ae875c8689530f1eb0
+Subproject commit fb49cea4932191a61ee541fa7ae1492ec6023d36
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 05667154..4df402c7 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 0566715473d62ada1a29728a51a9be6ff952e314
+Subproject commit 4df402c79236515bef110da033e322904360aaaf
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index 029b616a..42dfdc56 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -538,6 +538,11 @@ function! s:Gsubbr() abort
 endfunction
 
 function! s:Copy(...) abort
+    if &modified
+        echom 'Please check and save your file first!!!'
+        return 0
+    endif
+
     let newFile = (a:0 >= 1) ? a:1 : expand("%:p") . '.bak'
     exec '!rm ' . newFile
     exec 'w ' . newFile
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 9a974dbd..12788667 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -77,8 +77,16 @@ function! VFilter()
 endfunction
 
 function! ShowRemember()
-    let @" = expand('%:p')
-    echom expand('%:p')
+    let b:csdbpath = Cd2ProjectRoot("files.proj")
+    let relativePath = substitute(system('realpath --relative-to="' . b:csdbpath . '" ' . expand('%:p')), '\n', '', '')
+
+    if &clipboard == 'unnamed'
+        let @* = relativePath
+    else
+        let @+ = relativePath
+    endif
+
+    echom relativePath
     call SendTextToPbCopy(expand('%:p'))
 endfunction
 
