diff --git a/bashrc/generate_rsync_files.sh b/bashrc/generate_rsync_files.sh
index 7d7c91aa..d9adfb40 100755
--- ./bashrc/generate_rsync_files.sh
+++ ./bashrc/generate_rsync_files.sh
@@ -14,6 +14,12 @@ do
     or="-o"
 done < "$PRUNE_POSTFIX"
 
+if [ ${#include_params[@]} -gt 0 ]
+then
     find . "(" "${include_params[@]}" ")" -type f -size -9000k > "$rsyncFiles".diff && \
         comm -23 <(sort "$rsyncFiles") <(sort "$rsyncFiles".diff) > "$rsyncFiles".tmp && \
         cp -fv "$rsyncFiles".tmp rsync.files
+
+else
+    cp -fv "$rsyncFiles" rsync.files
+fi
diff --git a/vimrc/.vim/bundle/vim-buffergator b/vimrc/.vim/bundle/vim-buffergator
--- ./vimrc/.vim/bundle/vim-buffergator
+++ ./vimrc/.vim/bundle/vim-buffergator
@@ -1 +1 @@
-Subproject commit 6658fb7d238b0fcc04ee6543a495111ad49765ee
+Subproject commit 6658fb7d238b0fcc04ee6543a495111ad49765ee-dirty
