diff --git a/imvurc/generate.sh b/imvurc/generate.sh
index 7c49d05c..c278d081 100755
--- ./imvurc/generate.sh
+++ ./imvurc/generate.sh
@@ -1,11 +1,12 @@
 #!/bin/zsh
-file="$1"
+
+file=$(realpath "$1")
 target="$1.sh"
 
 if [ ! -f "$target" ]
 then
     cp -nv ~/loadrc/imvurc/generate_sh.template "$target"
-    rootFolder=$(~/loadrc/bashrc/find_up_folder.sh "$PWD/$file" "files.proj")
+    rootFolder=$(~/loadrc/bashrc/find_up_folder.sh "$file" "files.proj")
     relative_path=$(realpath --relative-to="$rootFolder" "$file")
     relative_path=$(echo "$relative_path" | sed 's/\//\\\//g')
     sed -i.bak "s/relative_path_var/$relative_path/g" "$target"
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index da3b1d57..489b5697 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit da3b1d57c57e96b3e107500cec675e4e2eba0373
+Subproject commit 489b56976f054cb883907bf6395b19c40e195eb8
