diff --git a/bashrc/generate_files_proj.sh b/bashrc/generate_files_proj.sh
index 5f5f2924..83e2c748 100755
--- ./bashrc/generate_files_proj.sh
+++ ./bashrc/generate_files_proj.sh
@@ -52,7 +52,7 @@ find . "(" "${prune_params[@]}" ")" -a -prune -o -size +0 -type f -exec grep -Il
     then \
         find . "(" "${include_params[@]}" ")" -type f -size -9000k | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' >> ${TARGET} ; \
         fi && \
-        echo "$(git rev-parse --abbrev-ref HEAD).gdio.diff" | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' >> ${TARGET} && \
+        echo "$(~/loadrc/gitrc/get_current_branch.sh).gdio.diff" | sed 's/\(["'\''\]\)/\\\1/g;s/.*/"&"/' >> ${TARGET} && \
         sort -u "$TARGET" -o "$TARGET" && \
         sed -i.bak 's/ /\\ /g' "$TARGET" && \
         cp -fv "$TARGET" files.proj && \
diff --git a/gitrc/reapply.sh b/gitrc/reapply.sh
index 978f4284..fc600652 100755
--- ./gitrc/reapply.sh
+++ ./gitrc/reapply.sh
@@ -1,6 +1,6 @@
 #!/bin/zsh
 DIFF="$1"
-git apply --reverse "$DIFF" 
+git apply --reverse "$(~/loadrc/gitrc/get_current_branch.sh).gdio.diff" 
 git checkout -- "$DIFF"
 git apply --reject --whitespace=fix "$DIFF"
 ~/loadrc/gitrc/gdio.sh
