diff --git a/imvurc/generate.sh b/imvurc/generate.sh
index 7c49d05c..c278d081 100755
--- ./imvurc/generate.sh
+++ ./imvurc/generate.sh
@@ -1,11 +1,12 @@
 #!/bin/zsh
-file="$1"
+
+file=$(realpath "$1")
 target="$1.sh"
 
 if [ ! -f "$target" ]
 then
     cp -nv ~/loadrc/imvurc/generate_sh.template "$target"
-    rootFolder=$(~/loadrc/bashrc/find_up_folder.sh "$PWD/$file" "files.proj")
+    rootFolder=$(~/loadrc/bashrc/find_up_folder.sh "$file" "files.proj")
     relative_path=$(realpath --relative-to="$rootFolder" "$file")
     relative_path=$(echo "$relative_path" | sed 's/\//\\\//g')
     sed -i.bak "s/relative_path_var/$relative_path/g" "$target"
diff --git a/vishrc/cat_run.sh b/vishrc/cat_run.sh
index 6cdef0d2..ddda55fe 100755
--- ./vishrc/cat_run.sh
+++ ./vishrc/cat_run.sh
@@ -2,5 +2,6 @@
 
 while read ss
 do
+    ss=$(echo "$ss" | sed 's/"//g;s/^\.\///g')
     ~/loadrc/vishrc/vrun.sh "$ss" &
 done < "$1"
