diff --git a/bashrc/generate_files_proj.sh b/bashrc/generate_files_proj.sh
index 71ea722d..2c184654 100755
--- ./bashrc/generate_files_proj.sh
+++ ./bashrc/generate_files_proj.sh
@@ -15,6 +15,8 @@ then
     exit 1
 fi
 
+mkdir -p .git
+
 TARGETEDIR=`realpath "$PWD"`
 cp -nv ~/loadrc/prunefix_template.conf ./prunefix.conf
 cp -nv ~/loadrc/prunefile_template.conf ./prunefile.conf
diff --git a/vimrc/.vim/bundle/tlib_vim b/vimrc/.vim/bundle/tlib_vim
index a071b6d4..70c4e222 160000
--- ./vimrc/.vim/bundle/tlib_vim
+++ ./vimrc/.vim/bundle/tlib_vim
@@ -1 +1 @@
-Subproject commit a071b6d41b20069a3520e0d101194a752968973b
+Subproject commit 70c4e222464020edc2809c932b488daaf891eeef
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 5892c774..17974649 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 5892c7748b0bafc68e9be9d2ce940b2e6e54939e
+Subproject commit 17974649e6f186c473e303aca3ad9dc7f223efea-dirty
