diff --git a/configure_gitk.sh b/configure_gitk.sh
new file mode 100755
index 00000000..f69e3bab
--- /dev/null
+++ ./configure_gitk.sh
@@ -0,0 +1,4 @@
+#!/bin/zsh
+
+GITK=$(realpath $(which gitk))
+sed -i.bak 's/set ignorespace 0/set ignorespace 1/g' "$GITK"  
diff --git a/install_basic.sh b/install_basic.sh
index 4e0529f9..78ee63e5 100755
--- ./install_basic.sh
+++ ./install_basic.sh
@@ -5,6 +5,7 @@ SCRIPT=$(realpath "$0") && \
 
 ./install_basic_linux.sh
 ./install_basic_macos.sh
+./configure_gitk.sh
 
 pip install \
     autoflake \
