diff --git a/gitrc/.globalgitignore b/gitrc/.globalgitignore
index b7d59f68..124d3757 100755
--- ./gitrc/.globalgitignore
+++ ./gitrc/.globalgitignore
@@ -69,4 +69,3 @@ setup.cfg
 prunefix.rsync
 rsync.files
 ncscopefile.*
-gbra.log
diff --git a/host_links/mini/etc/hosts b/host_links/mini/etc/hosts
index 892fab7b..aa7dd6d1 100644
--- ./host_links/mini/etc/hosts
+++ ./host_links/mini/etc/hosts
@@ -16,7 +16,6 @@
 172.27.68.119 imac
 172.27.118.58 vagrant
 192.168.31.99 mldonkey
-192.168.123.203 u2004
 
 # The following lines are desirable for IPv6 capable hosts
 ::1     ip6-localhost ip6-loopback
diff --git a/install_basic_macos.sh b/install_basic_macos.sh
index f3e2e567..ed72b1ad 100755
--- ./install_basic_macos.sh
+++ ./install_basic_macos.sh
@@ -24,7 +24,6 @@ brew install \
     hub \
     iperf \
     iperf3 \
-    jq \
     jupyterlab \
     kubectl \
     kubernetes-cli \
@@ -36,7 +35,6 @@ brew install \
     rename \
     tig \
     tmux \
-    w3m \
     wakeonlan \
     watch \
     wget \
diff --git a/kubernetesrc/install_helm.sh b/kubernetesrc/install_helm.sh
index 6569965a..ec429ec8 100755
--- ./kubernetesrc/install_helm.sh
+++ ./kubernetesrc/install_helm.sh
@@ -4,5 +4,5 @@ SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
 gofish install helm
-helm repo add stable https://kubernetes-charts.storage.googleapis.com
-helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com
+helm repo add stable https://charts.helm.sh/stable
+helm repo add incubator https://charts.helm.sh/incubator
diff --git a/linux/etc/ssh/ssh_config b/linux_links/etc/ssh/ssh_config
similarity index 100%
rename from linux/etc/ssh/ssh_config
rename to linux_links/etc/ssh/ssh_config
diff --git a/linux/etc/ssh/sshd_config b/linux_links/etc/ssh/sshd_config
similarity index 100%
rename from linux/etc/ssh/sshd_config
rename to linux_links/etc/ssh/sshd_config
diff --git a/macos/etc/sudoers b/macos/etc/sudoers
index 60b6bbcc..1148febe 100644
--- ./macos/etc/sudoers
+++ ./macos/etc/sudoers
@@ -44,5 +44,4 @@ root	ALL=(ALL) ALL
 # %users  ALL=/sbin/mount /cdrom,/sbin/umount /cdrom
 # %users  localhost=/sbin/shutdown -h now
 huangyingw     ALL=(ALL)       NOPASSWD: ALL
-yhuang     ALL=(ALL)       NOPASSWD: ALL
 admin     ALL=(ALL)       NOPASSWD: ALL
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index 8c6b7954..bf8e2a98 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit 8c6b79542fa875400bb03d82c8feed0d5fffc5d9
+Subproject commit bf8e2a98fa0eef2afb1b72b5af0495e6e6998e9a
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 768e6b6e..ff450cc6 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 768e6b6ec2b17273bd0cb943ece3451a5ff4bd2b
+Subproject commit ff450cc69fd0561dbaef1eac803b47df028c4d34
