diff --git a/bashrc/osmirror_exclude b/bashrc/osmirror_exclude
index cd9fe5f8..d00ebc3a 100644
--- ./bashrc/osmirror_exclude
+++ ./bashrc/osmirror_exclude
@@ -9,6 +9,7 @@
 */prunefix.conf
 boot/grub/grub.cfg
 etc/crypttab
+etc/exports
 etc/fstab
 etc/hostname
 etc/hosts
diff --git a/install_prerequisite.sh b/install_prerequisite.sh
index 298b6bc4..e055978f 100755
--- ./install_prerequisite.sh
+++ ./install_prerequisite.sh
@@ -41,6 +41,7 @@ apt-get install -y \
     mplayer \
     net-tools \
     netcat \
+    nfs-kernel-server \
     npm \
     openvpn \
     perl \
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 8468d6a8..a72f0ebe 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 8468d6a8a9dd53cf4cf1fa742b6d64cc440b2a62
+Subproject commit a72f0ebeaa2576c5ea967f7626bc6b3dbb0fa37f
