diff --git a/kubernetesrc/install_helm.sh b/kubernetesrc/install_helm.sh
index 6569965a..ec429ec8 100755
--- ./kubernetesrc/install_helm.sh
+++ ./kubernetesrc/install_helm.sh
@@ -4,5 +4,5 @@ SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
 gofish install helm
-helm repo add stable https://kubernetes-charts.storage.googleapis.com
-helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com
+helm repo add stable https://charts.helm.sh/stable
+helm repo add incubator https://charts.helm.sh/incubator
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index 8c6b7954..bf8e2a98 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit 8c6b79542fa875400bb03d82c8feed0d5fffc5d9
+Subproject commit bf8e2a98fa0eef2afb1b72b5af0495e6e6998e9a
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 768e6b6e..ff450cc6 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 768e6b6ec2b17273bd0cb943ece3451a5ff4bd2b
+Subproject commit ff450cc69fd0561dbaef1eac803b47df028c4d34
