diff --git a/bashrc/check_running.sh b/bashrc/check_running.sh
index 6ee8b3bf..638490ac 100755
--- ./bashrc/check_running.sh
+++ ./bashrc/check_running.sh
@@ -24,7 +24,7 @@ then
         # The process is not running
         # Echo current PID into lock file
         echo $$ > "${LCK_FILE}"
-        "$SCRIPT"
+        "$SCRIPT" "$2" "$3"
     else
         # the process IS running
         # handle it
diff --git a/leetcoderc/handle.sh b/leetcoderc/handle.sh
new file mode 100755
index 00000000..6397777b
--- /dev/null
+++ ./leetcoderc/handle.sh
@@ -0,0 +1,6 @@
+#!/bin/zsh
+FILE="$1"
+
+~/loadrc/leetcoderc/submit.sh "$FILE"
+NUM=$(echo "$FILE" | sed 's/.*\/\([0-9]*\)\..*/\1/g') 
+~/loadrc/bashrc/check_running.sh ~/loadrc/leetcoderc/submission.sh "$NUM"
diff --git a/leetcoderc/handle_test.sh b/leetcoderc/handle_test.sh
new file mode 100755
index 00000000..11541bd8
--- /dev/null
+++ ./leetcoderc/handle_test.sh
@@ -0,0 +1,6 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./handle.sh /Users/huangyingw/leetcode-cli/submissions/1.two-sum.260837994.Accepted.leetcode.py
diff --git a/leetcoderc/submission.sh b/leetcoderc/submission.sh
new file mode 100644
index 00000000..42d6a7e9
--- /dev/null
+++ ./leetcoderc/submission.sh
@@ -0,0 +1,3 @@
+#!/bin/zsh
+NUM="$1"
+leetcode submission "$NUM" -o ../downloads
diff --git a/leetcoderc/submit.sh b/leetcoderc/submit.sh
old mode 100755
new mode 100644
index 8653a009..df611c8a
--- ./leetcoderc/submit.sh
+++ ./leetcoderc/submit.sh
@@ -1,6 +1,3 @@
 #!/bin/zsh
-find ~/.lc -type f -name problems.json -delete
-find ~/myproject/git/interview/leetcode_solutions/ -type f -name "[0-9]*.py" | while read ss
-do
-    leetcode submit -d "$ss"
-done
+FILE="$1"
+leetcode submit "$FILE"
diff --git a/leetcoderc/submit_delete.sh b/leetcoderc/submit_delete.sh
new file mode 100755
index 00000000..8653a009
--- /dev/null
+++ ./leetcoderc/submit_delete.sh
@@ -0,0 +1,6 @@
+#!/bin/zsh
+find ~/.lc -type f -name problems.json -delete
+find ~/myproject/git/interview/leetcode_solutions/ -type f -name "[0-9]*.py" | while read ss
+do
+    leetcode submit -d "$ss"
+done
diff --git a/vishrc/vrun.sh b/vishrc/vrun.sh
index c56304f2..e14de500 100755
--- ./vishrc/vrun.sh
+++ ./vishrc/vrun.sh
@@ -10,7 +10,7 @@ fi
 
 if [[ "$file" = *'.leetcode.java' ]] || [[ "$file" = *'.leetcode.py' ]] || [[ "$file" = *'.leetcode.python3.py' ]]
 then
-    leetcode submit "$file"
+    ~/loadrc/leetcoderc/handle.sh "$file"
     exit 0
 fi
 
