diff --git a/neovim b/neovim
index 03e622e5..277694e5 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit 03e622e58686d209978eddf625a5f1ab7b69d491
+Subproject commit 277694e5e1777ff9ce35fc914cfaccdc27e26236-dirty
diff --git a/vimrc/.vim/bundle/neco-vim b/vimrc/.vim/bundle/neco-vim
index ec50b3cb..ba9b6535 160000
--- ./vimrc/.vim/bundle/neco-vim
+++ ./vimrc/.vim/bundle/neco-vim
@@ -1 +1 @@
-Subproject commit ec50b3cbb4537625c2aaba7601232a53ccd779aa
+Subproject commit ba9b6535381690fc6773d682fc046d8ddd2a863a
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 7b0b8fb5..a8d0cbeb 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 7b0b8fb54c39cf0027c356832b5b55283746eef6
+Subproject commit a8d0cbebf04cfe557abefbd0604a0613f518052c-dirty
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 78fb9e11..ecd31b60 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -507,12 +507,13 @@ set pastetoggle=<F3>            " when in insert mode, press <F3> to go to
 " open tig with Project root path
 nnoremap <Leader>t :TigOpenProjectRootDir<CR>
 nnoremap <leader>T :TigOpenCurrentFile<CR>
-nnoremap <silent> ml :call CopyLineInfo()<cr> 
+nnoremap <silent> ly :call CopyLineInfo()<cr> 
 
 function! CopyLineInfo()
+    normal yy
     let b:csdbpath = Cd2ProjectRoot("files.proj")
     let relativePath = substitute(system('realpath --relative-to="' . b:csdbpath . '" ' . expand('%:p')), '\n', '', '')
-    let content = relativePath . ':' . line('.') . ' ' . getline(line('.'))
+    let content = relativePath . ':' . line('.') . ' ' . @"
     let @+=content
 endfunction
 
