diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index a8d0cbeb..6b7caf0e 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit a8d0cbebf04cfe557abefbd0604a0613f518052c
+Subproject commit 6b7caf0e186cc0070a54a93833c43cd345945e81
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 78fb9e11..b39393e6 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -507,12 +507,13 @@ set pastetoggle=<F3>            " when in insert mode, press <F3> to go to
 " open tig with Project root path
 nnoremap <Leader>t :TigOpenProjectRootDir<CR>
 nnoremap <leader>T :TigOpenCurrentFile<CR>
-nnoremap <silent> ml :call CopyLineInfo()<cr> 
+nnoremap <silent> ly :call CopyLineInfo()<cr> 
 
 function! CopyLineInfo()
+    normal yy
     let b:csdbpath = Cd2ProjectRoot("files.proj")
     let relativePath = substitute(system('realpath --relative-to="' . b:csdbpath . '" ' . expand('%:p')), '\n', '', '')
-    let content = relativePath . ':' . line('.') . ' ' . getline(line('.'))
-    let @+=content
+    let content = relativePath . ':' . line('.') . ' ' . @"
+    let @*=content
 endfunction
 
