diff --git a/.gitignore b/.gitignore
index b98a32ce..8f92760e 100644
--- ./.gitignore
+++ ./.gitignore
@@ -1,5 +1,6 @@
 *.list
 *.log
+*.pkg
 .passwd
 bashrc/.bash_history
 bashrc/copy_files.log
diff --git a/zerotierrc/configure.sh b/zerotierrc/configure.sh
index 82ddf92f..724c11b3 100755
--- ./zerotierrc/configure.sh
+++ ./zerotierrc/configure.sh
@@ -7,9 +7,14 @@ deploy_configs() {
     TARGET="$1"
     SOURCE=~/loadrc/hosts/"`hostname`$TARGET"
 
+    if [ -L "$TARGET" ]
+    then
+        sudo rm "$TARGET"
+    fi
+
     if [ -f "$SOURCE" ]
     then
-        cp -fv --remove-destination "$SOURCE" "$TARGET"
+        sudo cp -fv "$SOURCE" "$TARGET"
     fi
 }
 
diff --git a/zerotierrc/install.sh b/zerotierrc/install.sh
index 349a4798..f1295fbc 100755
--- ./zerotierrc/install.sh
+++ ./zerotierrc/install.sh
@@ -7,4 +7,7 @@ if [ $(uname) != "Darwin" ]
 then
     curl -s https://install.zerotier.com/ | bash
     zerotier-cli join 93afae5963560e41
+else
+    wget -nc https://download.zerotier.com/dist/ZeroTier%20One.pkg -P ./installer/
+    sudo installer -package ./installer/ZeroTier\ One.pkg -target /
 fi
