diff --git a/bashrc/mirror.sh b/bashrc/mirror.sh
index 889ade9a..9d893982 100755
--- ./bashrc/mirror.sh
+++ ./bashrc/mirror.sh
@@ -12,7 +12,7 @@ TARGET=$2
 ready_file="$SOURCE"/"tmirror.ready"
 MIRRORCHECK=$HOME/loadrc/."`hostname`".mirror.check
 
-if [ ! -f ${ready_file} ]
+if [[ "${ready_file}" != *":"* ]] && [[ ! -f ${ready_file} ]]
 then
     echo -e "${red}tmirror must be run before mirror ... ${NC}"
     exit 1
