diff --git a/bashrc/osmirror_exclude b/bashrc/osmirror_exclude
index 5b16c128..cd9fe5f8 100644
--- ./bashrc/osmirror_exclude
+++ ./bashrc/osmirror_exclude
@@ -14,6 +14,7 @@ etc/hostname
 etc/hosts
 etc/libvirt
 etc/mdadm/mdadm.conf
+home/huangyingw/snap/
 media/dev
 media/nvme
 media/volgrp
diff --git a/wifirc/autoconnect.sh b/wifirc/autoconnect.sh
new file mode 100755
index 00000000..8929ce87
--- /dev/null
+++ ./wifirc/autoconnect.sh
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# find the uuid by nmcli con show 
+# and put this script, which is owned by root and executable, in the folder: /etc/NetworkManager/dispatcher.d
+
+if [ "$CONNECTION_UUID" = "df31faf5-3440-49ec-be88-9e040176663e" ]
+then
+    if [ "$2" = "down" ]; then
+        sleep 10
+        nmcli con up uuid $CONNECTION_UUID
+    fi
+fi
