diff --git a/bashrc/check_running.sh b/bashrc/check_running.sh
index 6ee8b3bf..638490ac 100755
--- ./bashrc/check_running.sh
+++ ./bashrc/check_running.sh
@@ -24,7 +24,7 @@ then
         # The process is not running
         # Echo current PID into lock file
         echo $$ > "${LCK_FILE}"
-        "$SCRIPT"
+        "$SCRIPT" "$2" "$3"
     else
         # the process IS running
         # handle it
diff --git a/bashrc/move.sh b/bashrc/move.sh
index fa3eff73..84be1e2f 100755
--- ./bashrc/move.sh
+++ ./bashrc/move.sh
@@ -19,7 +19,8 @@ realpath "$2" && TARGET=$(realpath "$2")
 
 if [ -n "$SOURCE" ] && [ -n "$TARGET" ] && [ "$SOURCE" != "$TARGET" ]
 then
-    rsync --remove-source-files -aHSv --progress --force "$SOURCE/" "$TARGET/"
+    rsync --remove-source-files -aHSv --progress --force "$SOURCE/" "$TARGET/" ; \
+        ~/loadrc/bashrc/rmEmpty.sh "$SOURCE/"
 else
     echo -e "${red}same dir --> ${SOURCE} ... ${NC}"
     echo -e "${red}please choose the different dir! ... ${NC}"
diff --git a/error b/error
index 83f1cc91..03ac2d1f 100644
--- ./error
+++ ./error
@@ -67,3 +67,5 @@ The legacy SnipMate parser is deprecated. Please see :h SnipMate-deprecate.
 [ncm2_tmux@yarp] subprocess.TimeoutExpired: Command '['tmux', 'display-message', '-p', '#{session_id}']' timed out after 5 seconds
 vimrc/.vim/bundle/ncm2-tmux/pythonx/ncm2_tmux.py:18:        proc = Popen(args=['tmux', 'display-message', '-p', '#{session_id}'],
 [ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh ...xcomplete' -p '^.' -s 'words' -l '-a' -c '' -g ''"]' died with <Signals.SIGKILL: 9>.
+[ncm2_tern@yarp] Job is dead. cmd=['/usr/local/opt/python@3.9/bin/python3.9', '-u',...', '/var/folders/7z/251_ly552l9glm39cnj9zxrh0000gn/T/nvimGXqrxd/0', 13, 'ncm2_tern']
+[ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh ...xcomplete' -p '^.' -s 'words' -l '-a' -c '' -g ''"]' died with <Signals.SIGKILL: 9>.
diff --git a/gitrc/.globalgitignore b/gitrc/.globalgitignore
index b7d59f68..628bec20 100755
--- ./gitrc/.globalgitignore
+++ ./gitrc/.globalgitignore
@@ -70,3 +70,4 @@ prunefix.rsync
 rsync.files
 ncscopefile.*
 gbra.log
+*.js_eslint_tmp_*.js
diff --git a/leetcoderc/remove_dup.sh b/leetcoderc/remove_dup.sh
index 5ba1dea0..dce2e372 100755
--- ./leetcoderc/remove_dup.sh
+++ ./leetcoderc/remove_dup.sh
@@ -2,6 +2,7 @@
 NUM="$1"
 cd "$NUM"
 
+git checkout .
 ~/loadrc/leetcoderc/remove_gabages.sh
 find . -type f \( -name \*.orig -o -name \*.bak \) -delete
 jdupes -1dNr .
diff --git a/leetcoderc/submit.sh b/leetcoderc/submit.sh
index 8653a009..df611c8a 100755
--- ./leetcoderc/submit.sh
+++ ./leetcoderc/submit.sh
@@ -1,6 +1,3 @@
 #!/bin/zsh
-find ~/.lc -type f -name problems.json -delete
-find ~/myproject/git/interview/leetcode_solutions/ -type f -name "[0-9]*.py" | while read ss
-do
-    leetcode submit -d "$ss"
-done
+FILE="$1"
+leetcode submit "$FILE"
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index f58203cb..5fd4a230 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -144,6 +144,10 @@ function! VRun()
     silent exec '!~/loadrc/bashrc/deploy.sh 2>&1 | tee deploy.findresult'
     call RunShell('~/loadrc/vishrc/vrun.sh', b:to_run, b:output)
 
+    if (expand("%") =~ '.*leetcode.*') 
+        call asyncrun#run('<bang>', '', '~/loadrc/leetcoderc/post_submit.sh ' . '"' .  b:file_name . '"' . ' 2>&1 | tee post_submit.log')
+    endif
+
     if b:to_run != 'gbil.log'
         call OpenOrSwitch(b:output, 'vs')
     else
@@ -151,7 +155,6 @@ function! VRun()
     endif
 
     call asyncrun#run('<bang>', '', '~/loadrc/bashrc/update_proj.sh') 
-    call asyncrun#run('<bang>', '', '~/loadrc/bashrc/deploy.sh 2>&1 | tee deploy.findresult')
 endfunction
 
 function! SearchAgain()
diff --git a/vishrc/vrun.sh b/vishrc/vrun.sh
index c56304f2..e14de500 100755
--- ./vishrc/vrun.sh
+++ ./vishrc/vrun.sh
@@ -10,7 +10,7 @@ fi
 
 if [[ "$file" = *'.leetcode.java' ]] || [[ "$file" = *'.leetcode.py' ]] || [[ "$file" = *'.leetcode.python3.py' ]]
 then
-    leetcode submit "$file"
+    ~/loadrc/leetcoderc/handle.sh "$file"
     exit 0
 fi
 
