diff --git a/vimrc/.vim/plugin/common.vim b/vimrc/.vim/plugin/common.vim
index 00796bc9..66850f7b 100644
--- ./vimrc/.vim/plugin/common.vim
+++ ./vimrc/.vim/plugin/common.vim
@@ -148,7 +148,7 @@ function! OpenOrSwitch(buffername, openMode)
     elseif a:openMode ==? "goto"
         silent exec 'e ' . a:buffername
     else
-        silent exec 'vs ' . a:buffername
+        silent exec 'topleft vs ' . a:buffername
     endif
 endfunction
 
