diff --git a/bashrc/tosmirror.sh b/bashrc/tosmirror.sh
index ec0dc3a6..efffa320 100755
--- ./bashrc/tosmirror.sh
+++ ./bashrc/tosmirror.sh
@@ -22,4 +22,3 @@ fi
 rsync -aHSvn --progress --delete-during --force \
     --exclude-from="$HOME"/loadrc/bashrc/osmirror_exclude \
     "${SOURCE}/" "${TARGET}/" > ~/loadrc/tosmirror.ready
-nvim ~/loadrc/tosmirror.ready  
diff --git a/error b/error
index 03ac2d1f..ceecba51 100644
--- ./error
+++ ./error
@@ -69,3 +69,4 @@ vimrc/.vim/bundle/ncm2-tmux/pythonx/ncm2_tmux.py:18:        proc = Popen(args=['
 [ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh ...xcomplete' -p '^.' -s 'words' -l '-a' -c '' -g ''"]' died with <Signals.SIGKILL: 9>.
 [ncm2_tern@yarp] Job is dead. cmd=['/usr/local/opt/python@3.9/bin/python3.9', '-u',...', '/var/folders/7z/251_ly552l9glm39cnj9zxrh0000gn/T/nvimGXqrxd/0', 13, 'ncm2_tern']
 [ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh ...xcomplete' -p '^.' -s 'words' -l '-a' -c '' -g ''"]' died with <Signals.SIGKILL: 9>.
+[ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh '/Users/huangyingw/loadrc/vimrc/.vim/bundle/tmux-complete.vim/sh/tmuxcomplete' -p '^.' -s
diff --git a/host_links/mini/etc/NetworkManager/system-connections/Millennium Flats b/host_links/mini/etc/NetworkManager/system-connections/Millennium Flats
deleted file mode 100644
index 923c4eb3..00000000
--- ./host_links/mini/etc/NetworkManager/system-connections/Millennium Flats	
+++ /dev/null
@@ -1,26 +0,0 @@
-[connection]
-id=Millennium Flats
-uuid=78278347-477b-4ecb-b62f-0e937ac6c83d
-type=wifi
-permissions=
-timestamp=1553749289
-
-[wifi]
-mac-address=AC:7B:A1:4E:65:29
-mac-address-blacklist=
-mode=infrastructure
-seen-bssids=F2:9F:C2:CA:3A:55;
-ssid=Millennium Flats
-
-[wifi-security]
-key-mgmt=wpa-psk
-psk=RedwoodCity
-
-[ipv4]
-dns-search=
-method=auto
-
-[ipv6]
-addr-gen-mode=stable-privacy
-dns-search=
-method=auto
diff --git a/host_links/server/etc/NetworkManager/system-connections/Millennium Flats b/host_links/server/etc/NetworkManager/system-connections/Millennium Flats
deleted file mode 100644
index c2c1543c..00000000
--- ./host_links/server/etc/NetworkManager/system-connections/Millennium Flats	
+++ /dev/null
@@ -1,24 +0,0 @@
-[connection]
-id=Millennium Flats
-uuid=b99bafe1-8232-47f9-b795-e79a5af50d39
-type=wifi
-permissions=
-
-[wifi]
-mac-address=B0:EA:BC:09:20:C6
-mac-address-blacklist=
-mode=infrastructure
-ssid=Millennium Flats
-
-[wifi-security]
-key-mgmt=wpa-psk
-psk=RedwoodCity
-
-[ipv4]
-dns-search=
-method=auto
-
-[ipv6]
-addr-gen-mode=stable-privacy
-dns-search=
-method=auto
diff --git a/host_links/server/etc/NetworkManager/system-connections/eth0 b/host_links/server/etc/NetworkManager/system-connections/eth0
deleted file mode 100644
index f865f7e2..00000000
--- ./host_links/server/etc/NetworkManager/system-connections/eth0
+++ /dev/null
@@ -1,25 +0,0 @@
-[connection]
-id=eth0
-uuid=5a0b787f-9027-4869-8dc5-013737405029
-type=ethernet
-interface-name=eth0
-permissions=
-timestamp=1589425127
-
-[ethernet]
-mac-address=10:7b:44:f0:e7:2f
-mac-address-blacklist=
-
-[ipv4]
-address1=192.168.3.2/24
-dns-priority=100
-dns-search=
-method=manual
-
-[ipv6]
-addr-gen-mode=stable-privacy
-dns-priority=100
-dns-search=
-method=ignore
-
-[proxy]
diff --git a/host_links/toshiba/etc/NetworkManager/system-connections/Millennium Flats b/host_links/toshiba/etc/NetworkManager/system-connections/Millennium Flats
deleted file mode 100644
index 0a32ae07..00000000
--- ./host_links/toshiba/etc/NetworkManager/system-connections/Millennium Flats	
+++ /dev/null
@@ -1,26 +0,0 @@
-[connection]
-id=Millennium Flats
-uuid=92862a87-cb1f-4e2d-b168-94b90d0097da
-type=wifi
-permissions=
-timestamp=1554784444
-
-[wifi]
-mac-address=40:25:C2:63:7E:0C
-mac-address-blacklist=
-mode=infrastructure
-seen-bssids=F0:9F:C2:C9:3A:49;
-ssid=Millennium Flats
-
-[wifi-security]
-key-mgmt=wpa-psk
-psk=RedwoodCity
-
-[ipv4]
-dns-search=
-method=auto
-
-[ipv6]
-addr-gen-mode=stable-privacy
-dns-search=
-method=auto
diff --git a/hosts/mini/etc/NetworkManager/system-connections/Millennium Flats b/hosts/mini/etc/NetworkManager/system-connections/Millennium Flats
deleted file mode 120000
index 63f39ffa..00000000
--- ./hosts/mini/etc/NetworkManager/system-connections/Millennium Flats	
+++ /dev/null
@@ -1 +0,0 @@
-/root/loadrc/host_links/mini/etc/NetworkManager/system-connections/Millennium Flats
\ No newline at end of file
diff --git a/install_basic.sh b/install_basic.sh
index 78ee63e5..aeaf8ba1 100755
--- ./install_basic.sh
+++ ./install_basic.sh
@@ -6,6 +6,7 @@ SCRIPT=$(realpath "$0") && \
 ./install_basic_linux.sh
 ./install_basic_macos.sh
 ./configure_gitk.sh
+./swith-2-zsh.sh
 
 pip install \
     autoflake \
diff --git a/swith-2-zsh.sh b/swith-2-zsh.sh
new file mode 100755
index 00000000..541ae968
--- /dev/null
+++ ./swith-2-zsh.sh
@@ -0,0 +1,9 @@
+#!/bin/zsh
+
+ZSH=$(which zsh)
+
+if [ "$ZSH" != "$SHELL" ]
+then
+    chsh -s "$ZSH"
+    reboot
+fi
diff --git a/vimrc/.vim/bundle/vim-buffergator b/vimrc/.vim/bundle/vim-buffergator
index 8d2ff564..87060211 160000
--- ./vimrc/.vim/bundle/vim-buffergator
+++ ./vimrc/.vim/bundle/vim-buffergator
@@ -1 +1 @@
-Subproject commit 8d2ff5644ea3bcfccc36b75d66c6ffd287922b1f
+Subproject commit 8706021133931d88cc38eafc01d4055d9c9bda9e
diff --git a/vimrc/.vim/plugin/common.vim b/vimrc/.vim/plugin/common.vim
index 00796bc9..66850f7b 100644
--- ./vimrc/.vim/plugin/common.vim
+++ ./vimrc/.vim/plugin/common.vim
@@ -148,7 +148,7 @@ function! OpenOrSwitch(buffername, openMode)
     elseif a:openMode ==? "goto"
         silent exec 'e ' . a:buffername
     else
-        silent exec 'vs ' . a:buffername
+        silent exec 'topleft vs ' . a:buffername
     endif
 endfunction
 
diff --git a/wifirc/autoconnect.sh b/wifirc/autoconnect.sh
index 60e018a5..e8d338a2 100755
--- ./wifirc/autoconnect.sh
+++ ./wifirc/autoconnect.sh
@@ -9,7 +9,7 @@
 # done
 
 
-if [ "$CONNECTION_UUID" = "df31faf5-3440-49ec-be88-9e040176663e" ]
+if [ "$CONNECTION_UUID" = "1c853a09-cf79-4a54-86fa-9efb5e447ea8" ]
 then
     if [ "$2" = "down" ]; then
         sleep 10
