diff --git a/.config/nvim/init.vim b/.config/nvim/init.vim
index f182e5b2..63ce3c0f 100644
--- ./.config/nvim/init.vim
+++ ./.config/nvim/init.vim
@@ -1,3 +1,4 @@
 set runtimepath^=~/.vim runtimepath+=~/.vim/after
 let &packpath = &runtimepath
 source ~/.vimrc
+lua require('plugins')
diff --git a/.gitmodules b/.gitmodules
index f74cb003..cf669a0c 100644
--- ./.gitmodules
+++ ./.gitmodules
@@ -233,3 +233,6 @@
 [submodule "neovim"]
 	path = neovim
 	url = git@github.com:huangyingw/neovim.git
+[submodule "vimrc/.vim/bundle/orgmode.nvim"]
+	path = vimrc/.vim/bundle/orgmode.nvim
+	url = git@github.com:kristijanhusak/orgmode.nvim.git
diff --git a/neovim b/neovim
index 5931cffd..fb2dbb91 160000
--- ./neovim
+++ ./neovim
@@ -1 +1 @@
-Subproject commit 5931cffd46149d011ff40a49758ed8afcf47f9d7
+Subproject commit fb2dbb917235871a6bb0d4f49017dda60a91fa0d
diff --git a/vimrc/.vim/bundle/orgmode.nvim b/vimrc/.vim/bundle/orgmode.nvim
new file mode 160000
index 00000000..385dd9ae
--- /dev/null
+++ ./vimrc/.vim/bundle/orgmode.nvim
@@ -0,0 +1 @@
+Subproject commit 385dd9ae82dd0eadb02ed6e07df133670edb2271
