diff --git a/.envrc b/.envrc
index 47a567de..50f5b8e2 100755
--- ./.envrc
+++ ./.envrc
@@ -27,3 +27,4 @@ export red='\033[0;31m'
 export PATH="$HOME/anaconda3/bin:$PATH"
 export PATH=/usr/local/bin:$PATH
 export PATH=~/bin:$PATH
+export PATH=/opt/homebrew/bin:$PATH
diff --git a/.loadrc b/.loadrc
index 17e45232..bd41d7ab 100755
--- ./.loadrc
+++ ./.loadrc
@@ -1,4 +1,5 @@
 #!/bin/zsh
+. $HOME/loadrc/.pathrc
 . $HOME/loadrc/.androidrc
 . $HOME/loadrc/.dockerrc
 . $HOME/loadrc/.envrc
diff --git a/.pathrc b/.pathrc
new file mode 100644
index 00000000..75068b27
--- /dev/null
+++ ./.pathrc
@@ -0,0 +1,11 @@
+export PATH="$HOME/Android/sdk/platform-tools:$PATH"
+export PATH="$HOME/Android/sdk/tools:$PATH"
+export PATH="$HOME/anaconda3/bin:$PATH"
+export PATH="/Applications/NoMachine.app/Contents/MacOS:$PATH"
+export PATH="/Applications/kdiff3.app/Contents/MacOS:$PATH"
+export PATH="/usr/local/bin:/opt/local/sbin:$PATH"
+export PATH=$PATH:/usr/local/go/bin
+export PATH=/opt/homebrew/bin:$PATH
+export PATH=/usr/local/bin:$PATH
+export PATH=/usr/local/cuda/bin:$PATH
+export PATH=~/bin:$PATH
diff --git a/.scl-1yihuang-m.corporate.localrc b/.scl-1yihuang-m.corporate.localrc
deleted file mode 100755
index 13bad558..00000000
--- ./.scl-1yihuang-m.corporate.localrc
+++ /dev/null
@@ -1,60 +0,0 @@
-########
-# TOMCAT
-export TOMCAT_HOME=/Users/yihuang/bin/apache-tomcat-7.0.55
-export PATH=${TOMCAT_HOME}/bin:${PATH}
-
-######
-# JAVA
-export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.7.0_67.jdk/Contents/Home/
-export PATH=${JAVA_HOME}/bin:${PATH}
-export PATH=/opt/subversion/bin:${PATH}
-
-#######
-# MAVEN
-export M2_HOME=/Users/yihuang/bin/apache-maven-3.2.3/
-export PATH=${M2_HOME}/bin:${PATH}
-
-export MANPATH=${JAVA_HOME}/man:${MANPATH}
-
-#######
-# MYSQL
-export MYSQL_HOME=/usr/local/mysql
-export PATH=${MYSQL_HOME}/bin:${PATH}
-
-
-########
-# GRADLE
-export GRADLE_HOME=/Users/yihuang/bin/gradle-1.12
-export PATH=${GRADLE_HOME}/bin:${PATH}
-
-
-###########
-# LIQUIBASE
-export LIQUIBASE_HOME=/Users/yihuang/bin/liquibase-3.2.0
-export PATH=${LIQUIBASE_HOME}:${PATH}
-
-
-#######
-#MONGOD
-
-export SVN_EDITOR=/usr/bin/vim
-
-######
-# PATH
-export MY_HOME=/Users/yihuang
-export PATH=${MY_HOME}/bin:${PATH}
-export PATH=$HOME/local/bin:$PATH
-
-#################
-# COLOR DIRECTORY
-export CLICOLOR=1
-export LSCOLORS=ExFxCxDxBxegedabagacad
-
-#######
-# ALIAS
-#alias ls='ls -lAF --color=tty'
-#alias ls='ls -la'
-alias updatedb='sudo /usr/libexec/locate.updatedb'
-alias runmongo='mongod --fork --logpath /var/log/mongodb.log'
-
-export PATH=/usr/local/Cellar/ctags/5.8/bin:${PATH}
diff --git a/bashrc/mosh.sh b/bashrc/mosh.sh
index d62867f4..b3ad0742 100755
--- ./bashrc/mosh.sh
+++ ./bashrc/mosh.sh
@@ -2,18 +2,12 @@
 target=$(echo "$1" | sed  -e "s/\/$//g")
 mkdir "$target"
 
-if ( ssh "$target" "uname" | grep -q Darwin )
-then
-    MOSHSERVER="/usr/local/bin/mosh-server"
-    tmuxAction='/usr/local/bin/tmux new-session -A -s mosh'
-else
-    MOSHSERVER="/usr/bin/mosh-server"
-    tmuxAction='/usr/bin/tmux new-session -A -s mosh'
-fi
+MOSHSERVER=$(ssh "$target" ". ~/loadrc/.loadrc ; which mosh-server")
+tmuxAction="$(ssh "$target" ". ~/loadrc/.loadrc ; which tmux") new-session -A -s mosh"
 
 while true
 do
     ssh -fnN -R 2224:localhost:2224 "$target" &
-    ssh "$target" "~/loadrc/macosrc/configure_mosh.sh" &
+    ssh "$target" ". ~/loadrc/.loadrc ; ~/loadrc/macosrc/configure_mosh.sh" &
     mosh --server="$MOSHSERVER" "$target" -- ${tmuxAction}
 done
diff --git a/bashrc/ssh.sh b/bashrc/ssh.sh
index 7042dce9..6b78b3a0 100755
--- ./bashrc/ssh.sh
+++ ./bashrc/ssh.sh
@@ -1,11 +1,4 @@
 #! /bin/zsh
 target=$(echo "$1" | sed  -e "s/\/$//g")
-
-if ( ssh "$target" "uname" | grep -q Darwin )
-then
-    tmuxAction='/usr/local/bin/tmux new-session -A -s ssh'
-else
-    tmuxAction='/usr/bin/tmux new-session -A -s ssh'
-fi
-
+tmuxAction="$(ssh "$target" ". ~/loadrc/.loadrc ; which tmux") new-session -A -s ssh"
 ssh -Y "$target" -t -- ${tmuxAction}
diff --git a/macosrc/configure_mosh.sh b/macosrc/configure_mosh.sh
index bb27cd34..a45fa3fe 100755
--- ./macosrc/configure_mosh.sh
+++ ./macosrc/configure_mosh.sh
@@ -1,5 +1,5 @@
 #!/bin/zsh
-BINPATH=$(/usr/local/bin/realpath $(which mosh-server))
+BINPATH=$(realpath $(which mosh-server))
 
 # Turn the firewall off
 sudo /usr/libexec/ApplicationFirewall/socketfilterfw --setglobalstate off
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index dbe8b1e2..5bb65f40 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit dbe8b1e21e987866bde27a008deda5d0d767bee3
+Subproject commit 5bb65f409e3fd1222e9fbba59c17ded0b458595f
