diff --git a/vimrc/.vim/plugin/pbpaste.vim b/vimrc/.vim/plugin/pbpaste.vim
index 40f455b1..c6303a7a 100644
--- ./vimrc/.vim/plugin/pbpaste.vim
+++ ./vimrc/.vim/plugin/pbpaste.vim
@@ -1,7 +1,8 @@
 function! PopulatePasteBufferFromOSX()
   let @+ = system('pbpaste-remote')
-  normal! "+p
+  normal! gvp
+  
 endfunction
   
 nnoremap <silent> cp :call PopulatePasteBufferFromOSX()<cr>
-vmap <silent> cp :call PopulatePasteBufferFromOSX()<cr>
+vnoremap <silent> cp :call PopulatePasteBufferFromOSX()<cr>
