diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index af437c98..c1bb7b1c 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -77,7 +77,12 @@ function! VFilter()
 endfunction
 
 function! ShowRemember()
-    let @+=expand('%:p')
+    if &clipboard == 'unnamed'
+        let @* =expand('%:p')
+    else
+        let @@ =expand('%:p')
+    endif
+
     echom expand('%:p')
 endfunction
 
