diff --git a/.macmini.rc b/.macmini.rc
deleted file mode 100644
index 61ca121c..00000000
--- ./.macmini.rc
+++ /dev/null
@@ -1,7 +0,0 @@
-#!/bin/zsh
-#alias restartWifi='$HOME/loadrc/osxrc/restart_wifi.sh'
-
-if ! (sudo systemsetup -getcomputersleep | grep -q Never)
-then
-    sudo systemsetup -setcomputersleep Never
-fi
diff --git a/.mini.rc b/.mini.rc
new file mode 100755
index 00000000..3d146d68
--- /dev/null
+++ ./.mini.rc
@@ -0,0 +1,2 @@
+#!/bin/zsh
+~/loadrc/setup_pbpaste-remote.sh
diff --git a/bashrc/mosh.sh b/bashrc/mosh.sh
index bb6e35fe..5ad03da2 100755
--- ./bashrc/mosh.sh
+++ ./bashrc/mosh.sh
@@ -14,6 +14,7 @@ fi
 while true
 do
     ssh -fnN -R 2224:localhost:2224 "$target" &
+    ssh -fnN -R 2225:localhost:2225 "$target" &
     ssh "$target" ". ~/loadrc/.pathrc ; ~/loadrc/macosrc/configure_mosh.sh" &
     mosh --server="$MOSHSERVER" "$target" -- ${tmuxAction}
 done
diff --git a/gitrc/checkout_rejs.sh b/gitrc/checkout_rejs.sh
index ddbfb613..5b7555ad 100755
--- ./gitrc/checkout_rejs.sh
+++ ./gitrc/checkout_rejs.sh
@@ -3,7 +3,13 @@
 for ss in $(git status | grep \.rej$) ; \
 do \
     targetFile=$(echo "$ss" | sed 's/\.rej$//g')
+    if [ "$2" = "f" ]
+    then
+        git checkout "$1" "$targetFile" && \
+            rm "$ss"
+    else
         git show "$1":"$targetFile" > "$ss"
+    fi
 done
 
 git show "$1":COMMIT_EDITMSG > COMMIT_EDITMSG
diff --git a/gitrc/gcom.sh b/gitrc/gcom.sh
index 83a83235..556cdc79 100755
--- ./gitrc/gcom.sh
+++ ./gitrc/gcom.sh
@@ -23,5 +23,3 @@ if [ -n "$clean" ]
 then
     git stash pop stash@{0}
 fi
-
-~/loadrc/gitrc/gdio.sh
diff --git a/gitrc/gdi2.sh b/gitrc/gdi2.sh
index 2b769ee7..6754ec3c 100755
--- ./gitrc/gdi2.sh
+++ ./gitrc/gdi2.sh
@@ -38,10 +39,11 @@ fi
     git apply --index --reject --whitespace=fix "$GDITDIFF"
 
 retVal=$?
+
 if [ $retVal -ne 0 ]
 then
-    ~/loadrc/gitrc/checkout_rejs.sh "$currentBranch"
-else
+    ~/loadrc/gitrc/checkout_rejs.sh "$currentBranch" "$1"
+elif [ $retVal -eq 0 ] || [ "$1" = "f" ]
     git commit  --no-verify -am "$commit_message" && \
         git pull ; \
         git push ; \
diff --git a/gitrc/gpl.sh b/gitrc/gpl.sh
index 9cd20920..4f53bd55 100755
--- ./gitrc/gpl.sh
+++ ./gitrc/gpl.sh
@@ -2,7 +2,7 @@
 
 ~/loadrc/gitrc/include_gitconfig.sh
 git remote update &
-git checkout *.gdio.diff
+~/loadrc/gitrc/discard_unnecessaries.sh
 git pull &
 currentBranch=$(~/loadrc/gitrc/get_current_branch.sh)
 
diff --git a/install_basic_macos.sh b/install_basic_macos.sh
index 94af8469..a26c9986 100755
--- ./install_basic_macos.sh
+++ ./install_basic_macos.sh
@@ -12,10 +12,6 @@ fi
 git -C $(brew --repository homebrew/core) checkout master
 brew update
 brew upgrade
-brew install \
-    mosh \
-    tmux
-
 brew install \
     Caskroom/cask/xquartz \
     astyle \
@@ -38,9 +34,11 @@ brew install \
     libtool \
     lynx \
     minikube \
+    mosh \
     openssl \
     rename \
     tig \
+    tmux \
     w3m \
     wakeonlan \
     watch \
diff --git a/pbpaste-remote b/pbpaste-remote
new file mode 100755
index 00000000..80d3b345
--- /dev/null
+++ ./pbpaste-remote
@@ -0,0 +1,2 @@
+#!/bin/sh
+nc localhost 2225
diff --git a/setup_pbpaste-remote.sh b/setup_pbpaste-remote.sh
new file mode 100755
index 00000000..5351f6c9
--- /dev/null
+++ ./setup_pbpaste-remote.sh
@@ -0,0 +1,11 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0") && \
+    SCRIPTPATH=$(dirname "$SCRIPT") && \
+    cd "$SCRIPTPATH" || cd ~/loadrc/
+
+if [ -f ~/bin/pbpaste-remote ]
+then
+    exit 0
+fi
+
+cp -fv ./pbpaste-remote ~/bin/pbpaste-remote
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 17ceb981..c2611960 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 17ceb981ac27663a704beaf9a59eea259bab46e3
+Subproject commit c26119605885fcc878023b2faa8dcdb2cde89761
diff --git a/vimrc/.vim/plugin/diff-options.vim b/vimrc/.vim/plugin/diff-options.vim
new file mode 100644
index 00000000..a154e243
--- /dev/null
+++ ./vimrc/.vim/plugin/diff-options.vim
@@ -0,0 +1,4 @@
+set diffopt+=iwhiteall
+set diffopt+=iblank
+set diffopt+=algorithm:patience
+set diffopt+=indent-heuristic
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index cac3f5f2..f53036fd 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -425,7 +425,7 @@ function! s:Gdi2(...) abort
     let worktree = Cd2Worktree()
     let output = 'gdi2.findresult'
     let arg1 = (a:0 >= 1) ? a:1 : ''
-    exec '!~/loadrc/gitrc/gdi2.sh 2>&1 | tee ' . '"' .  output . '"'
+    exec '!~/loadrc/gitrc/gdi2.sh ' . '"' .  arg1 . '"' . ' 2>&1 | tee ' . '"' .  output . '"'
     call OpenOrSwitch(output, 'vs')
 endfunction
 
diff --git a/vimrc/.vim/plugin/pbpaste.vim b/vimrc/.vim/plugin/pbpaste.vim
new file mode 100644
index 00000000..f21d88a8
--- /dev/null
+++ ./vimrc/.vim/plugin/pbpaste.vim
@@ -0,0 +1,8 @@
+function! PopulatePasteBufferFromOSX()
+  " let @+ = system('nc localhost 2225')
+  let @+ = system('pbpaste-remote')
+  echom @+
+  echo "done"
+endfunction
+
+nnoremap <silent> cp :call PopulatePasteBufferFromOSX()<cr>
diff --git a/vimrc/.vim/plugin/view.vim b/vimrc/.vim/plugin/view.vim
index 9ee20986..a7bd3f54 100644
--- ./vimrc/.vim/plugin/view.vim
+++ ./vimrc/.vim/plugin/view.vim
@@ -37,9 +37,4 @@ else
     set winwidth=999999
 endif
 
-set diffopt+=iwhiteall
-set diffopt+=iblank
-set diffopt+=algorithm:patience
-set diffopt+=indent-heuristic
 set wrap
-"set iskeyword-=_
