diff --git a/install_nodejs.sh b/install_nodejs.sh
index c3036dee..34cb2b71 100755
--- ./install_nodejs.sh
+++ ./install_nodejs.sh
@@ -5,3 +5,9 @@ cd "$SCRIPTPATH"
 
 . ~/.zshrc
 nvm install node
+OTHER_NODE="/usr/local/bin/node"
+ 
+if [ -f "$OTHER_NODE" ]
+then
+    mv -v "$OTHER_NODE" "$OTHER_NODE".bak
+fi
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index c2611960..773c1669 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit c26119605885fcc878023b2faa8dcdb2cde89761
+Subproject commit 773c1669a071d0553b0cc2c93049d9c07ce418f3
diff --git a/vimrc/.vim/plugin/pbpaste.vim b/vimrc/.vim/plugin/pbpaste.vim
index 40f455b1..80fa0585 100644
--- ./vimrc/.vim/plugin/pbpaste.vim
+++ ./vimrc/.vim/plugin/pbpaste.vim
@@ -1,7 +1,12 @@
-function! PopulatePasteBufferFromOSX()
+function! PopulatePasteBufferFromOSX(mode)
     let @+ = system('pbpaste-remote')
+    if a:mode== 'v'
+        normal! gvp
+    else
         normal! "+p
+    endif
+
 endfunction
 
-nnoremap <silent> cp :call PopulatePasteBufferFromOSX()<cr>
-vmap <silent> cp :call PopulatePasteBufferFromOSX()<cr>
+nnoremap <silent> cp :call PopulatePasteBufferFromOSX('n')<cr>
+vnoremap <silent> cp :call PopulatePasteBufferFromOSX('v')<cr>
