diff --git a/gitrc/reapply.sh b/gitrc/reapply.sh
index 0b291d09..7d9e0081 100755
--- ./gitrc/reapply.sh
+++ ./gitrc/reapply.sh
@@ -3,3 +3,4 @@ DIFF="$1"
 git apply --reverse "$DIFF" 
 git checkout -- "$DIFF"
 git apply --reject --whitespace=fix "$DIFF"
+~/loadrc/gitrc/gdio.sh
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index e6d0ed1a..2b8c1ae2 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -666,6 +666,8 @@ endfunction
 function! s:Reapply() abort
     let worktree = Cd2Worktree()
     exec '!~/loadrc/gitrc/reapply.sh ' . '"' .  expand("%:p") . '"'
+    e
+    call s:Gdi()
 endfunction
 
 function! s:Split() abort
