diff --git a/gitrc/reapply.sh b/gitrc/reapply.sh
index 0b291d09..978f4284 100755
--- ./gitrc/reapply.sh
+++ ./gitrc/reapply.sh
@@ -3,3 +3,5 @@ DIFF="$1"
 git apply --reverse "$DIFF" 
 git checkout -- "$DIFF"
 git apply --reject --whitespace=fix "$DIFF"
+~/loadrc/gitrc/gdio.sh
+git add "$DIFF"
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index e6d0ed1a..e969fa0e 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -664,8 +664,15 @@ function! s:Gfix() abort
 endfunction
 
 function! s:Reapply() abort
+    if (expand("%") !~ '.*gdio.diff')
+        echom 'Please only run on *gdio.diff!!!'
+        return 0
+    endif
+
     let worktree = Cd2Worktree()
     exec '!~/loadrc/gitrc/reapply.sh ' . '"' .  expand("%:p") . '"'
+    call s:Gs()
+    call s:Gdi()
 endfunction
 
 function! s:Split() abort
