diff --git a/dockerrc/configure.sh b/dockerrc/configure.sh
deleted file mode 100755
index f2fc63a9..00000000
--- ./dockerrc/configure.sh
+++ /dev/null
@@ -1,8 +0,0 @@
-#!/bin/zsh
-SCRIPT=$(realpath "$0")
-SCRIPTPATH=$(dirname "$SCRIPT")
-cd "$SCRIPTPATH"
-
-mkdir -p /media/nvme/var/lib/docker/
-service docker stop
-~/loadrc/bashrc/mklnk.sh /var/lib/docker/ /media/nvme/var/lib/docker/
diff --git a/kvmrc/configure.sh b/kvmrc/configure.sh
deleted file mode 100755
index 926a191a..00000000
--- ./kvmrc/configure.sh
+++ /dev/null
@@ -1,9 +0,0 @@
-#!/bin/zsh
-SCRIPT=$(realpath "$0")
-SCRIPTPATH=$(dirname "$SCRIPT")
-cd "$SCRIPTPATH"
-
-mkdir -p /media/nvme/var/lib/libvirt/
-mkdir -p /media/nvme/etc/libvirt/
-~/loadrc/bashrc/mklnk.sh /var/lib/libvirt/ /media/nvme/var/lib/libvirt/
-~/loadrc/bashrc/mklnk.sh /etc/libvirt/ /media/nvme/etc/libvirt/
