diff --git a/bashrc/generate_files_from.sh b/bashrc/generate_files_from.sh
new file mode 100755
index 00000000..3c0c7ef0
--- /dev/null
+++ ./bashrc/generate_files_from.sh
@@ -0,0 +1,22 @@
+#!/bin/zsh
+rsyncFiles=rsync.files.bak
+cat files.proj | sed 's/^"//g;s/"$//g;s/\\ / /g' > "$rsyncFiles"
+PRUNE_POSTFIX=prunefix.rsync
+
+if [ ! -f "$PRUNE_POSTFIX" ]
+then
+    exit
+fi
+
+or="";
+
+while read suf
+do
+    suf=$(echo "$suf" | sed 's/"//g')
+    include_params+=( $or "-wholename" "$suf" )
+    or="-o"
+done < "$PRUNE_POSTFIX"
+
+find . "(" "${include_params[@]}" ")" -type f -size -9000k > "$rsyncFiles".tmp && \
+    comm -23 <(sort "$rsyncFiles") <(sort "$rsyncFiles".tmp) > "$rsyncFiles" && \
+    cp -fv "$rsyncFiles" rsync.files
diff --git a/bashrc/rsyncFiles.sh b/bashrc/rsyncFiles.sh
index 647be6cf..57ae30ab 100755
--- ./bashrc/rsyncFiles.sh
+++ ./bashrc/rsyncFiles.sh
@@ -2,8 +2,7 @@
 host=$(git config deploy.host)
 rpath=$(git config deploy.path)
 
-cat files.proj | sed 's/^"//g;s/"$//g;s/\\ / /g' > files.proj.tmp && \
 rsync -aHv --force --progress \
-    --files-from files.proj.tmp \
+    --files-from rsync.files \
     . \
     "$host:$rpath"
diff --git a/generate_files_from.test.sh b/generate_files_from.test.sh
new file mode 100755
index 00000000..36284323
--- /dev/null
+++ ./generate_files_from.test.sh
@@ -0,0 +1,6 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./bashrc/generate_files_from.sh
diff --git a/gitrc/dogsync.sh b/gitrc/dogsync.sh
index f1a0f55f..25554679 100755
--- ./gitrc/dogsync.sh
+++ ./gitrc/dogsync.sh
@@ -1,3 +1,4 @@
 #!/bin/zsh
-~/loadrc/gitrc/gpl.sh && ~/loadrc/gitrc/dogdio.sh &
-~/loadrc/gitrc/gps.sh &
+~/loadrc/gitrc/gpl.sh
+~/loadrc/gitrc/gps.sh
+~/loadrc/gitrc/dogdio.sh
diff --git a/gitrc/gpl.sh b/gitrc/gpl.sh
index 5aa5d719..9b5215c0 100755
--- ./gitrc/gpl.sh
+++ ./gitrc/gpl.sh
@@ -7,17 +7,18 @@ currentBranch=$(~/loadrc/gitrc/get_current_branch.sh)
 
 for ss in `git remote -v |awk '/\(fetch\)$/{print $1}'`
 do
-    git pull $ss $currentBranch
+    git pull $ss $currentBranch &
+    git remote prune $ss &
 done
 
 for ss in $(git config --get-all pull.from)
 do
     remote_branch=$(echo "$ss" | sed  -e "s/\// /g")
-    COMMAND="git pull $remote_branch"
+    COMMAND="git pull $remote_branch &"
     eval "$COMMAND"
 done
 
 if [ -n $(git config gsync.remote) ]
 then
-    git pull $(git config gsync.remote) $(git config gsync.branch)
+    git pull $(git config gsync.remote) $(git config gsync.branch) &
 fi
diff --git a/gitrc/gps.sh b/gitrc/gps.sh
index 4e584f27..2198cb7f 100755
--- ./gitrc/gps.sh
+++ ./gitrc/gps.sh
@@ -6,6 +6,6 @@ if [[ $(git config --get-all push.remote) ]]
 then
     for ss in $(git config --get-all push.remote)
     do
-        git push --all -v "$ss" && git push -v --tag "$ss"
+        git push --all -v "$ss" && git push -v --tag "$ss" &
     done
 fi
diff --git a/prunefix.rsync b/prunefix.rsync
new file mode 100644
index 00000000..d85a2356
--- /dev/null
+++ ./prunefix.rsync
@@ -0,0 +1,2 @@
+"./.git/modules/*"
+"./host_links/*"
