diff --git a/.gitignore b/.gitignore
index f4b46900..b98a32ce 100644
--- ./.gitignore
+++ ./.gitignore
@@ -10,6 +10,7 @@ host_links/ubuntu/home/parallels/.config/smplayer/smplayer.ini
 kvmrc/.ropeproject/
 tmirror.ready
 tosmirror.ready
+var/
 vimrc/.vim/.netrwhist
 vimrc/.vim/autosess/
 vimrc/.vim/spell/en.utf-8.add.spl
diff --git a/bashrc/rsyncFiles.sh b/bashrc/rsyncFiles.sh
index 57ae30ab..36367474 100755
--- ./bashrc/rsyncFiles.sh
+++ ./bashrc/rsyncFiles.sh
@@ -6,3 +6,8 @@ rsync -aHv --force --progress \
     --files-from rsync.files \
     . \
     "$host:$rpath"
+
+rsync -aHv --force --progress \
+    --files-from=files.rev \
+    "$host:/" \
+    .
diff --git a/vishrc/vrun.sh b/vishrc/vrun.sh
index 8d5a77f3..987db2c5 100755
--- ./vishrc/vrun.sh
+++ ./vishrc/vrun.sh
@@ -56,10 +56,6 @@ case $extension in
         if [[ -n "$host" ]] && [[ "$host" != "localhost" ]]
         then
             ssh -nY "$host" "cd $rpath ; . ~/loadrc/.loadrc ; $rpath/$rfile"
-            rsync -aHv --force --progress \
-                --files-from=files.rev \
-                "$host:$rpath/" \
-                .
         else
             . ~/loadrc/.loadrc
             "$file"
