diff --git a/dockerrc/dps.sh b/dockerrc/dps.sh
index 6c5385fa..be034a79 100755
--- ./dockerrc/dps.sh
+++ ./dockerrc/dps.sh
@@ -1,2 +1,2 @@
 #!/bin/zsh
-docker ps > dps.findresult
+docker ps
diff --git a/gitrc/.globalgitignore b/gitrc/.globalgitignore
index a9a4f4a0..3a3f3dd2 100755
--- ./gitrc/.globalgitignore
+++ ./gitrc/.globalgitignore
@@ -2,8 +2,7 @@
 *.bak
 *.bin
 *.diff
-!*.fix.gdio.diff
-!dev.gdio.diff
+!*.gdio.diff
 *.exe
 *.findresult
 *.fugitiveblame
diff --git a/vimrc/.vim/bundle/python-mode b/vimrc/.vim/bundle/python-mode
index c2a1b140..40406e03 160000
--- ./vimrc/.vim/bundle/python-mode
+++ ./vimrc/.vim/bundle/python-mode
@@ -1 +1 @@
-Subproject commit c2a1b14045959245b4cc3beece1d87844dd07917
+Subproject commit 40406e03dd0fc91b906d3d566c318082c152c3ed
diff --git a/vimrc/.vim/plugin/common.vim b/vimrc/.vim/plugin/common.vim
index 87bbd96d..adddcfad 100644
--- ./vimrc/.vim/plugin/common.vim
+++ ./vimrc/.vim/plugin/common.vim
@@ -211,24 +211,22 @@ function! VsMax(fileName)
     call OpenOrSwitch(a:fileName, 'vs')
 endfunc
 
-function! RunShell(shell, ...)
-    let arg1 = (a:0 >= 1) ? a:1 : ''
-    let arg2 = (a:0 >= 2) ? a:2 : ''
-    let silent = substitute(system('git config vrun.silent'), '\n', '', '')
-    let async = substitute(system('git config vrun.async'), '\n', '', '')
-    let temp_log = arg2 . '.findresult'
-    let run_string = a:shell . ' ' . '"' .  arg1 . '"' .  ' ' . '2>&1 | tee' . ' ' . temp_log
-
-    if async ==? "true"
+function! RunShell(shell, param, output, async)
+    if &modified
+        echom 'Please check and save your file first!!!'
+        return 0
+    endif
+
+    let temp_log = output . '.findresult'
+    let run_string = a:shell . ' ' . '"' .  param . '"' .  ' ' . '2>&1 | tee' . ' ' . temp_log
+
+    if async ==? 'true'
         call asyncrun#run('<bang>', '', ' ' . run_string)
-    else
-        if silent ==? "true"
-            silent exec '!' . run_string
     else
         exec '!' . run_string
     endif
-    endif
-    silent exec '!cp' . ' ' . '"' .  temp_log . '"' . ' ' . '"' .  arg2 . '"'
+
+    silent exec '!cp' . ' ' . '"' .  temp_log . '"' . ' ' . '"' .  output . '"'
 endfunc
 
 function! Filter2Findresult()
diff --git a/vimrc/.vim/plugin/fugitive.vim b/vimrc/.vim/plugin/fugitive.vim
index f339827f..346073f8 100644
--- ./vimrc/.vim/plugin/fugitive.vim
+++ ./vimrc/.vim/plugin/fugitive.vim
@@ -37,10 +37,8 @@ command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gfix :
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gicb :execute s:Gicb()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gitk :execute s:Gitk(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Glf :execute s:Glf()
-command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Glg :execute s:Glg()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gmet :execute s:Gmet()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gpl :execute s:Gpl()
-command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gps :execute s:Gps()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Grsh :execute s:Grsh(<q-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Grta :execute s:Grta(<f-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Grtu :execute s:Grtu()
@@ -52,8 +50,6 @@ command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gst :e
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gsti :execute s:Gsti()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gstl :execute s:Gstl()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gstlv :execute s:Gstlv()
-command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gstp :execute s:Gstp(<q-args>)
-command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gstv :execute s:Gstv(<q-args>)
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gsync :execute s:Gsync()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gtg :execute s:Gtg()
 command! -bang -bar -nargs=* -complete=customlist,fugitive#CompleteObject Gkd :execute s:Gkd(<f-args>)
@@ -163,27 +159,6 @@ function! s:Glf() abort
     call OpenOrSwitch('glf.findresult', 'vs')
 endfunction
 
-function! s:Glg() abort
-    let worktree = Cd2Worktree()
-    silent exec '!~/loadrc/gitrc/glg.sh'
-    call OpenOrSwitch('glg.findresult', 'vs')
-endfunction
-
-function! s:Gps() abort
-    let worktree = Cd2Worktree()
-    call asyncrun#run('<bang>', '', '~/loadrc/gitrc/gps.sh 2>&1 | tee gps.findresult')
-endfunction
-
-function! s:Gstp(args, ...) abort
-    let worktree = Cd2Worktree()
-    call asyncrun#run('<bang>', '', '~/loadrc/gitrc/gstp.sh ' . '"' .  a:args . '"')
-endfunction
-
-function! s:Gstv(args, ...) abort
-    let worktree = Cd2Worktree()
-    call asyncrun#run('<bang>', '', '~/loadrc/gitrc/gstv.sh ' . '"' .  a:args . '"')
-endfunction
-
 function! s:Gcof(...) abort
     let worktree = Cd2Worktree()
     let b:relativePath = substitute(expand('%:p'), worktree . '/', "", "g")
@@ -516,14 +491,9 @@ function! s:Copy(...) abort
 endfunction
 
 function! s:CatPlay(...) abort
-    if &modified
-        echom 'Please check and save your file first!!!'
-        return 0
-    endif
-
     call asyncrun#stop('<bang>')
     let b:output = expand("%:p") . '.runresult'
-    call asyncrun#run('<bang>', '', '~/loadrc/vishrc/cat_play.sh ' . '"' . expand("%:p") . '"' . ' 2>&1 | tee ' . b:output)
+    call RunShell('~/loadrc/vishrc/cat_play.sh', expand("%:p"), b:output, 'true')
     call OpenOrSwitch(b:output, 'vs')
 endfunction
 
@@ -539,24 +509,14 @@ function! s:CatMove(...) abort
 endfunction
 
 function! s:CatDu(...) abort
-    if &modified
-        echom 'Please check and save your file first!!!'
-        return 0
-    endif
-
     let b:output = expand("%:p") . '.runresult'
-    call RunShell('~/loadrc/vishrc/cat_du.sh', expand("%:p"), b:output)
+    call RunShell('~/loadrc/vishrc/cat_du.sh', expand("%:p"), b:output, 'true') 
     call OpenOrSwitch(b:output, 'vs')
 endfunction
 
 function! s:CatRun(...) abort
-    if &modified
-        echom 'Please check and save your file first!!!'
-        return 0
-    endif
-
     let b:output = expand("%:p") . '.runresult'
-    call RunShell('~/loadrc/vishrc/cat_run.sh', expand("%:p"), b:output)
+    call RunShell('~/loadrc/vishrc/cat_run.sh', expand("%:p"), b:output, 'true') 
     call OpenOrSwitch(b:output, 'vs')
 endfunction
 
@@ -566,8 +526,9 @@ endfunction
 
 function! s:Dps() abort
     let worktree = Cd2Worktree()
-    silent exec '!~/loadrc/dockerrc/dps.sh'
-    call OpenOrSwitch('dps.findresult', 'vs')
+    let b:output = 'dps.findresult'
+    call RunShell('~/loadrc/dockerrc/dps.sh', '', b:output, '')
+    call OpenOrSwitch(b:output, 'vs')
 endfunction
 
 function! s:SvnUp() abort
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index fcfe1713..c6285abb 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -110,7 +110,7 @@ function! VDebug()
 
     let b:csdbpath = Cd2ProjectRoot("files.proj")
     let b:output = b:csdbpath . '/' . b:file_name . '.runresult'
-    call RunShell('~/loadrc/vishrc/vdebug.sh', b:to_run, b:output)
+    call RunShell('~/loadrc/vishrc/vdebug.sh', b:to_run, b:output, 'true') 
 
     if b:to_run != 'gbil.log'
         call OpenOrSwitch(b:output, 'vs')
@@ -139,7 +139,7 @@ function! VRun()
 
     let b:csdbpath = Cd2ProjectRoot("files.proj")
     let b:output = b:csdbpath . '/' . b:file_name . '.runresult'
-    call RunShell('~/loadrc/vishrc/vrun.sh', b:to_run, b:output)
+    call RunShell('~/loadrc/vishrc/vrun.sh', b:to_run, b:output, 'true') 
 
     if b:to_run != 'gbil.log'
         call OpenOrSwitch(b:output, 'vs')
@@ -163,9 +163,9 @@ function! GetFirstColumnOfFile()
         return 0
     endif
 
-    let bak_file = substitute(system('~/loadrc/bashrc/get_first_column_of_file.sh ' . '"' .  expand("%:p") . '"'), '\n', '', '')
-    silent exec '!~/loadrc/bashrc/get_first_column_of_file.sh ' . '"' .  expand("%:p") . '"'
-    call OpenOrSwitch(expand("%:p") . '.bak', 'vs')
+    let b:output = expand("%:p") . '.bak'
+    call RunShell('~/loadrc/bashrc/get_first_column_of_file.sh', expand("%:p"), b:output, '') 
+    call OpenOrSwitch(b:output, 'vs')
 endfunction
 
 function! CSCSearch(num)
