diff --git a/gitrc/.gitconfig b/gitrc/.gitconfig
index ca91dbaf..eda62d25 100755
--- ./gitrc/.gitconfig
+++ ./gitrc/.gitconfig
@@ -90,3 +90,5 @@
     prompt = false
 [merge "ours"]
 	driver = true
+[credential "https://github.com"]
+	helper = !gh auth git-credential
diff --git a/install_basic_linux.sh b/install_basic_linux.sh
index 34269226..7370103f 100755
--- ./install_basic_linux.sh
+++ ./install_basic_linux.sh
@@ -10,6 +10,7 @@ then
 fi
 
 ./install_prerequisite.sh
+./install_gh.sh
 ./install_nodejs.sh
 ./install_time_machine.sh
 ./enpass_install.sh
diff --git a/install_gh.sh b/install_gh.sh
new file mode 100755
index 00000000..c3363697
--- /dev/null
+++ ./install_gh.sh
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg |  gpg --dearmor -o /usr/share/keyrings/githubcli-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" |  tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+apt update
+apt install gh
