diff --git a/.envrc b/.envrc
index 47a567de..50f5b8e2 100755
--- ./.envrc
+++ ./.envrc
@@ -27,3 +27,4 @@ export red='\033[0;31m'
 export PATH="$HOME/anaconda3/bin:$PATH"
 export PATH=/usr/local/bin:$PATH
 export PATH=~/bin:$PATH
+export PATH=/opt/homebrew/bin:$PATH
diff --git a/bashrc/mosh.sh b/bashrc/mosh.sh
index d62867f4..b3ad0742 100755
--- ./bashrc/mosh.sh
+++ ./bashrc/mosh.sh
@@ -2,18 +2,12 @@
 target=$(echo "$1" | sed  -e "s/\/$//g")
 mkdir "$target"
 
-if ( ssh "$target" "uname" | grep -q Darwin )
-then
-    MOSHSERVER="/usr/local/bin/mosh-server"
-    tmuxAction='/usr/local/bin/tmux new-session -A -s mosh'
-else
-    MOSHSERVER="/usr/bin/mosh-server"
-    tmuxAction='/usr/bin/tmux new-session -A -s mosh'
-fi
+MOSHSERVER=$(ssh "$target" ". ~/loadrc/.loadrc ; which mosh-server")
+tmuxAction="$(ssh "$target" ". ~/loadrc/.loadrc ; which tmux") new-session -A -s mosh"
 
 while true
 do
     ssh -fnN -R 2224:localhost:2224 "$target" &
-    ssh "$target" "~/loadrc/macosrc/configure_mosh.sh" &
+    ssh "$target" ". ~/loadrc/.loadrc ; ~/loadrc/macosrc/configure_mosh.sh" &
     mosh --server="$MOSHSERVER" "$target" -- ${tmuxAction}
 done
diff --git a/bashrc/ssh.sh b/bashrc/ssh.sh
index 7042dce9..6b78b3a0 100755
--- ./bashrc/ssh.sh
+++ ./bashrc/ssh.sh
@@ -1,11 +1,4 @@
 #! /bin/zsh
 target=$(echo "$1" | sed  -e "s/\/$//g")
-
-if ( ssh "$target" "uname" | grep -q Darwin )
-then
-    tmuxAction='/usr/local/bin/tmux new-session -A -s ssh'
-else
-    tmuxAction='/usr/bin/tmux new-session -A -s ssh'
-fi
-
+tmuxAction="$(ssh "$target" ". ~/loadrc/.loadrc ; which tmux") new-session -A -s ssh"
 ssh -Y "$target" -t -- ${tmuxAction}
diff --git a/install_prerequisite.sh b/install_prerequisite.sh
index e055978f..e2824b9e 100755
--- ./install_prerequisite.sh
+++ ./install_prerequisite.sh
@@ -8,7 +8,6 @@ apt-get install -y \
     astyle \
     automake \
     autotools-dev \
-    bcmwl-kernel-source \
     binutils \
     build-essential \
     cmake \
diff --git a/macosrc/configure_mosh.sh b/macosrc/configure_mosh.sh
index bb27cd34..a45fa3fe 100755
--- ./macosrc/configure_mosh.sh
+++ ./macosrc/configure_mosh.sh
@@ -1,5 +1,5 @@
 #!/bin/zsh
-BINPATH=$(/usr/local/bin/realpath $(which mosh-server))
+BINPATH=$(realpath $(which mosh-server))
 
 # Turn the firewall off
 sudo /usr/libexec/ApplicationFirewall/socketfilterfw --setglobalstate off
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index dbe8b1e2..5bb65f40 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit dbe8b1e21e987866bde27a008deda5d0d767bee3
+Subproject commit 5bb65f409e3fd1222e9fbba59c17ded0b458595f
