diff --git a/crontabrc/.imacpro.cron b/crontabrc/.imacpro.cron
new file mode 100644
index 00000000..c2a422cf
--- /dev/null
+++ ./crontabrc/.imacpro.cron
@@ -0,0 +1 @@
+* * * * * /Users/huangyingw/loadrc/crontabrc/setup_cron.sh
diff --git a/crontabrc/setup_cron.sh b/crontabrc/setup_cron.sh
new file mode 100644
index 00000000..a3890a2f
--- /dev/null
+++ ./crontabrc/setup_cron.sh
@@ -0,0 +1,2 @@
+#!/bin/zsh
+/Users/huangyingw/loadrc/setup.sh 2>&1 | tee /Users/huangyingw/loadrc/setup.log
