diff --git a/vimrc/.vim/ftdetect/swift.vim b/vimrc/.vim/ftdetect/swift.vim
new file mode 100644
index 00000000..5be9bf33
--- /dev/null
+++ ./vimrc/.vim/ftdetect/swift.vim
@@ -0,0 +1,12 @@
+autocmd BufNewFile,BufRead *.swift set filetype=swift
+autocmd BufRead * call s:Swift()
+function! s:Swift()
+  if !empty(&filetype)
+    return
+  endif
+
+  let line = getline(1)
+  if line =~ "^#!.*swift"
+    setfiletype swift
+  endif
+endfunction
