diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index 6c96a890..91f311f5 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit 6c96a89033dec2f30be9ec60d45de6844e59e327
+Subproject commit 91f311f5ff6f62ed7ed137d38afb386bf189550f-dirty
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 90057d91501256c25c56f2a1dfaf789b1eb4969a
+Subproject commit 90057d91501256c25c56f2a1dfaf789b1eb4969a-dirty
diff --git a/vimrc/.vim/plugin/auto_cscope.vim b/vimrc/.vim/plugin/auto_cscope.vim
index 38806d87..bbe09820 100644
--- ./vimrc/.vim/plugin/auto_cscope.vim
+++ ./vimrc/.vim/plugin/auto_cscope.vim
@@ -1,4 +1,8 @@
 function! s:auto_cscope()
+    if &buftype ==# 'terminal'
+        return
+    endif
+
     call UpdateProj()
 endfunction
 
diff --git a/vimrc/.vim/plugin/auto_deploy.vim b/vimrc/.vim/plugin/auto_deploy.vim
index 52ba2bd3..8b9c3418 100644
--- ./vimrc/.vim/plugin/auto_deploy.vim
+++ ./vimrc/.vim/plugin/auto_deploy.vim
@@ -1,4 +1,8 @@
 function! s:auto_deploy_augroup()
+    if &buftype ==# 'terminal'
+        return
+    endif
+
     call Cd2ProjectRoot('files.proj')
     call asyncrun#run('<bang>', '', '~/loadrc/bashrc/deploy.sh 2>&1 | tee deploy.findresult')
     call CHANGE_CURR_DIR() 
