diff --git a/etc/netatalk/afp.conf b/etc/netatalk/afp.conf
new file mode 100644
index 00000000..73b495da
--- /dev/null
+++ ./etc/netatalk/afp.conf
@@ -0,0 +1,3 @@
+[Time Machine]
+    path = target_dir_var
+    time machine = yes
diff --git a/install_basic_linux.sh b/install_basic_linux.sh
index 0b7a87e8..23be74d2 100755
--- ./install_basic_linux.sh
+++ ./install_basic_linux.sh
@@ -11,6 +11,7 @@ fi
 
 ./install_prerequisite.sh
 ./install_nodejs.sh
+./install_time_machine.sh
 ./enpass_install.sh
 ./nxrc/install.sh
 ./kubernetesrc/install.sh
diff --git a/install_time_machine.sh b/install_time_machine.sh
new file mode 100755
index 00000000..4af77eea
--- /dev/null
+++ ./install_time_machine.sh
@@ -0,0 +1,16 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+TARGET_DIR="/media/mldonkey/time-machine"
+TARGET_DIR_FOR_SED=$(echo "$TARGET_DIR" | sed 's/\//\\\//g')
+apt install -y netatalk avahi-daemon
+cp -fv ./etc/netatalk/afp.conf /etc/netatalk/afp.conf
+sed -i.bak "s/target_dir_var/$TARGET_DIR_FOR_SED/g" /etc/netatalk/afp.conf
+
+mkdir -p "$TARGET_DIR"
+chown nobody:nogroup "$TARGET_DIR"
+chmod -R 777 "$TARGET_DIR"
+
+service netatalk restart
