diff --git a/bashrc/update_proj.sh b/bashrc/update_proj.sh
index 1c2673c0..923ffff4 100755
--- ./bashrc/update_proj.sh
+++ ./bashrc/update_proj.sh
@@ -1,2 +1,9 @@
 #!/bin/zsh
+
+if [ ! -f files.proj ]
+then
+    echo -e "${red}No files.proj file here, will not update the proj ... ${NC}"
+    exit 1
+fi
+
 ~/loadrc/bashrc/check_running.sh ~/loadrc/bashrc/do_update_proj.sh
diff --git a/vimrc/.vim/plugin/auto_deploy.vim b/vimrc/.vim/plugin/auto_deploy.vim
index 52ba2bd3..181f7aa9 100644
--- ./vimrc/.vim/plugin/auto_deploy.vim
+++ ./vimrc/.vim/plugin/auto_deploy.vim
@@ -1,5 +1,10 @@
 function! s:auto_deploy_augroup()
     call Cd2ProjectRoot('files.proj')
+
+    if !filereadable('files.proj')
+        return
+    endif
+
     call asyncrun#run('<bang>', '', '~/loadrc/bashrc/deploy.sh 2>&1 | tee deploy.findresult')
     call CHANGE_CURR_DIR() 
 endfunction
