diff --git a/error b/error
index 3cae4633..467ed7fa 100644
--- ./error
+++ ./error
@@ -61,3 +61,7 @@ E15: Invalid expression: l:changed_win_bufs
 Cought anything: Vim(cd):E472: Command failed
 
 [ncm2_go@yarp] TypeError: expected str, bytes or os.PathLike object, not NoneType
+
+Error detected while processing function CSCSearch[12]..GetEscapedResult:
+line   15:
+E33: No previous substitute regular expression
diff --git a/pythonrc/parse_vtt.py b/pythonrc/parse_vtt.py
new file mode 100644
index 00000000..628ad2d8
--- /dev/null
+++ ./pythonrc/parse_vtt.py
@@ -0,0 +1,18 @@
+#!/usr/bin/env python3
+import datetime
+import re
+
+time_string = "00:00:08.690 --> 00:00:08.700 align:start position:0%"
+
+r2 = re.findall(r"\d+:\d+:\d+(?=[^:]*-->)", time_string)
+print(r2)
+
+date_time = datetime.datetime.strptime(r2[0], "%H:%M:%S")
+
+a_timedelta = date_time - datetime.datetime(1900, 1, 1)
+seconds = a_timedelta.total_seconds()
+
+print(seconds)
+
+def parse_vtt_str(vttStr):
+    return 8
diff --git a/pythonrc/time_string.py b/pythonrc/time_string.py
new file mode 100644
index 00000000..cec13115
--- /dev/null
+++ ./pythonrc/time_string.py
@@ -0,0 +1,12 @@
+#!/usr/bin/env python
+import datetime
+
+time_string = "01:01:09"
+date_time = datetime.datetime.strptime(time_string, "%H:%M:%S")
+
+print(date_time)
+
+a_timedelta = date_time - datetime.datetime(1900, 1, 1)
+seconds = a_timedelta.total_seconds()
+
+print(seconds)
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 75de359f..bb24924b 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -92,7 +92,7 @@ function! PlayVideo()
     endif
 
     call asyncrun#stop('<bang>')
-    call asyncrun#run('<bang>', '', '~/loadrc/vishrc/vlc.sh ' . '"' . expand('%:p:h') . '/' . '"' .  getline('.'))
+    call asyncrun#run('<bang>', '', '~/loadrc/vishrc/vlc.py ' . '"' . expand('%:p:h') . '"' .  ' ' . '"' . getline('.') . '"')
 endfunction
 
 function! VDebug()
diff --git a/vishrc/vlc.py b/vishrc/vlc.py
new file mode 100755
index 00000000..8e9687cc
--- /dev/null
+++ ./vishrc/vlc.py
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+from os import path
+import sys
+from pathlib import Path
+home = str(Path.home()) + '/'
+sys.path.insert(1, home + 'loadrc/pythonrc/')
+from parse_vtt import parse_vtt_str
+
+with open(home + "output.txt", "w") as text_file:
+    text_file.write("sys.argv[0] --> %s\n" % sys.argv[0])
+    text_file.write("sys.argv[1] --> %s\n" % sys.argv[1])
+    text_file.write("sys.argv[2] --> %s\n" % sys.argv[2])
+    fileVar = sys.argv[1]
+    lineVar = sys.argv[2]
+    text_file.write("fileVar --> %s\n" % fileVar)
+    text_file.write("lineVar --> %s\n" % lineVar)
+    targetFile = fileVar + '/' + lineVar
+    text_file.write("targetFile --> %s\n" % targetFile)
+    if path.exists(targetFile):
+        text_file.write("path.exists --> \n")
+        start = parse_vtt_str(lineVar)
+        text_file.write("start --> %s\n" % start)
+    else:
+        text_file.write("path.exists not --> \n")
diff --git a/vishrc/vlc.py.sh b/vishrc/vlc.py.sh
new file mode 100755
index 00000000..53e407bd
--- /dev/null
+++ ./vishrc/vlc.py.sh
@@ -0,0 +1,6 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./vlc.py "/Users/huangyingw/Dropbox/learning/AI/fast.ai" "02:11:12.620 --> 02:11:17.560"
