diff --git a/Library/Preferences/org.videolan.vlc/vlcrc b/Library/Preferences/org.videolan.vlc/vlcrc
index adb3ad5a..30134b49 100644
--- ./Library/Preferences/org.videolan.vlc/vlcrc
+++ ./Library/Preferences/org.videolan.vlc/vlcrc
@@ -1,5 +1,5 @@
 ﻿###
-###  vlc 3.0.11
+###  vlc 3.0.12
 ###
 
 ###
@@ -381,6 +381,35 @@ macosx-continue-playback=1
 # Verbosity (integer)
 #log-verbose=-1
 
+[rist] # RIST input
+
+# RIST maximum packet size (bytes) (integer)
+#packet-size=1472
+
+# RIST demux/decode maximum jitter (default is 5ms) (integer)
+#maximum-jitter=5
+
+# RIST latency (ms) (integer)
+#latency=1000
+
+# RIST nack retry interval (ms) (integer)
+#retry-interval=132
+
+# RIST reorder buffer (ms) (integer)
+#reorder-buffer=70
+
+# RIST maximum retry count (integer)
+#max-retries=10
+
+# RIST nack type, 0 = range, 1 = bitmask. Default is range (integer)
+#nack-type=0
+
+# Disable NACK output packets (boolean)
+#disable-nacks=0
+
+# Do not check for a valid rtcp message from the encoder (boolean)
+#mcast-blind-nacks=0
+
 [audiobargraph_v] # Audio Bar Graph Video sub source
 
 # X coordinate (integer)
@@ -522,7 +551,7 @@ macosx-continue-playback=1
 #sout-avi-subject=
 
 # Encoder (string)
-#sout-avi-encoder=VLC Media Player - 3.0.11 Vetinari
+#sout-avi-encoder=VLC Media Player - 3.0.12 Vetinari
 
 # Keywords (string)
 #sout-avi-keywords=
@@ -1111,6 +1140,23 @@ macosx-continue-playback=1
 # Sync on Intra Frame (boolean)
 #packetizer-mpegvideo-sync-iframe=0
 
+[access_output_rist] # RIST stream output
+
+# RIST target packet size (bytes) (integer)
+#sout-rist-packet-size=1328
+
+# RIST data output caching size (ms) (integer)
+#sout-rist-caching=50
+
+# RIST retry-buffer queue size (ms) (integer)
+#sout-rist-buffer-size=0
+
+# SSRC used in RTP output (default is random, i.e. 0) (integer)
+#sout-rist-ssrc=0
+
+# Stream name (string)
+#sout-rist-stream-name=
+
 [svcdsub] # Philips OGT (SVCD subtitle) decoder
 
 [ci_filters] # Mac OS X hardware video filters
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 768e6b6e..dba38728 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 768e6b6ec2b17273bd0cb943ece3451a5ff4bd2b
+Subproject commit dba38728e665c37417c6d37d62d1329b421973c7
