diff --git a/macos/Users/huangyingw/Library/Preferences/org.videolan.vlc/vlcrc b/macos/Users/huangyingw/Library/Preferences/org.videolan.vlc/vlcrc
index 342f83c4..8d5879d6 100644
--- ./macos/Users/huangyingw/Library/Preferences/org.videolan.vlc/vlcrc
+++ ./macos/Users/huangyingw/Library/Preferences/org.videolan.vlc/vlcrc
@@ -1,5 +1,5 @@
 ﻿###
-###  vlc 3.0.11
+###  vlc 3.0.11.1
 ###
 
 ###
@@ -522,7 +522,7 @@ macosx-continue-playback=1
 #sout-avi-subject=
 
 # Encoder (string)
-#sout-avi-encoder=VLC Media Player - 3.0.11 Vetinari
+#sout-avi-encoder=VLC Media Player - 3.0.11.1 Vetinari
 
 # Keywords (string)
 #sout-avi-keywords=
@@ -4089,13 +4089,13 @@ key-jump+medium=Up
 #global-key-jump-long=
 
 # Long backwards jump (string)
-#key-jump-long=Command+Shift+Alt+Left
+key-jump-long=k
 
 # Long forward jump (string)
 #global-key-jump+long=
 
 # Long forward jump (string)
-#key-jump+long=Command+Shift+Alt+Right
+key-jump+long=j
 
 # Next frame (string)
 #global-key-frame-next=
@@ -4191,7 +4191,7 @@ key-quit=n
 #global-key-subdelay-up=
 
 # Subtitle delay up (string)
-#key-subdelay-up=j
+key-subdelay-up=
 
 # Subtitle delay down (string)
 #global-key-subdelay-down=
diff --git a/vimrc/.vim/plugin/keymap.vim b/vimrc/.vim/plugin/keymap.vim
index 7f6cbd82..93e550e6 100644
--- ./vimrc/.vim/plugin/keymap.vim
+++ ./vimrc/.vim/plugin/keymap.vim
@@ -140,7 +140,7 @@ function! VRun()
         let b:to_run = b:to_run . '.sh'
     endif
 
-    let b:csdbpath = Cd2ProjectRoot("files.proj")
+    let b:csdbpath = Cd2Worktree()
     let b:output = b:csdbpath . '/' . b:file_name . '.runresult'
     call RunShell('~/loadrc/vishrc/vrun.sh', b:to_run, b:output)
 
diff --git a/vishrc/vrun.sh b/vishrc/vrun.sh
index 1ad765d3..f7654a88 100755
--- ./vishrc/vrun.sh
+++ ./vishrc/vrun.sh
@@ -57,7 +57,6 @@ case $extension in
         then
             ssh -nY "$host" "cd $rpath ; . ~/loadrc/.loadrc ; $rpath/$rfile"
         else
-            . ~/loadrc/.loadrc
             "$file"
         fi
         ;;
