diff --git a/bashrc/do_update_proj.sh b/bashrc/do_update_proj.sh
index cbbd6cc8..9a6e19c8 100755
--- ./bashrc/do_update_proj.sh
+++ ./bashrc/do_update_proj.sh
@@ -1,7 +1,7 @@
 #!/bin/zsh
 
 ~/loadrc/bashrc/generate_files_proj.sh
+~/loadrc/bashrc/generate_rsync_files.sh
 ~/loadrc/bashrc/fvideos.sh
 ~/loadrc/bashrc/fdocs.sh
 ~/loadrc/bashrc/cscope.sh
-~/loadrc/bashrc/generate_rsync_files.sh
diff --git a/host_links/server/etc/hosts b/host_links/server/etc/hosts
index 885df571..50886663 100644
--- ./host_links/server/etc/hosts
+++ ./host_links/server/etc/hosts
@@ -4,7 +4,7 @@
 192.168.31.111 u2004
 172.27.217.162 localhost.imvu.com
 192.168.31.140 mini
-192.168.31.149 minds
+192.168.123.132 minds
 192.168.31.153 gluster00
 192.168.31.129 gluster01
 192.168.31.115 gluster02
diff --git a/vimrc/.vim/bundle/vim-fugitive b/vimrc/.vim/bundle/vim-fugitive
index 90e8d986..e8cff84f 160000
--- ./vimrc/.vim/bundle/vim-fugitive
+++ ./vimrc/.vim/bundle/vim-fugitive
@@ -1 +1 @@
-Subproject commit 90e8d986e5341db18361a7dee3e621c93d633080
+Subproject commit e8cff84f1bd2fa551a229289042357c700661c4d
diff --git a/vishrc/vrun.sh b/vishrc/vrun.sh
index 1ad765d3..f7654a88 100755
--- ./vishrc/vrun.sh
+++ ./vishrc/vrun.sh
@@ -57,7 +57,6 @@ case $extension in
         then
             ssh -nY "$host" "cd $rpath ; . ~/loadrc/.loadrc ; $rpath/$rfile"
         else
-            . ~/loadrc/.loadrc
             "$file"
         fi
         ;;
