diff --git a/error b/error
index d2d4d1ca..5963196d 100644
--- ./error
+++ ./error
@@ -71,3 +71,4 @@ vimrc/.vim/bundle/ncm2-tmux/pythonx/ncm2_tmux.py:18:        proc = Popen(args=['
 [ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh ...xcomplete' -p '^.' -s 'words' -l '-a' -c '' -g ''"]' died with <Signals.SIGKILL: 9>.
 [ncm2_tmux_complete@yarp] subprocess.CalledProcessError: Command '['sh', '-c', "sh '/Users/huangyingw/loadrc/vimrc/.vim/bundle/tmux-complete.vim/sh/tmuxcomplete' -p '^.' -s
 [ncm2_tern@yarp] Job is dead. cmd=['/usr/local/opt/python@3.9/bin/python...ders/zn/6zwn3dz13v3fdjtrsh9g_dv9ymvzkm/T/nvimUC0moc/0', 13, 'ncm2_tern']
+[ncm2_jedi@yarp] Job is dead. cmd=['/root/anaconda3/bin/python3', '-u', '/root/.vim/bundle/nvim-yarp/pythonx/yarp.py', '/tmp/nvimEcygXB/0', 6, 'ncm2_jedi']
diff --git a/etc/netatalk/afp.conf b/etc/netatalk/afp.conf
index 73b495da..46d444a3 100644
--- ./etc/netatalk/afp.conf
+++ ./etc/netatalk/afp.conf
@@ -1,3 +1,3 @@
-[Time Machine]
+[netatalk]
     path = target_dir_var
     time machine = yes
diff --git a/install_time_machine.sh b/install_time_machine.sh
index 4af77eea..3022d17c 100755
--- ./install_time_machine.sh
+++ ./install_time_machine.sh
@@ -3,14 +3,12 @@ SCRIPT=$(realpath "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
-TARGET_DIR="/media/mldonkey/time-machine"
+TARGET_DIR="/"
 TARGET_DIR_FOR_SED=$(echo "$TARGET_DIR" | sed 's/\//\\\//g')
 apt install -y netatalk avahi-daemon
 cp -fv ./etc/netatalk/afp.conf /etc/netatalk/afp.conf
 sed -i.bak "s/target_dir_var/$TARGET_DIR_FOR_SED/g" /etc/netatalk/afp.conf
 
 mkdir -p "$TARGET_DIR"
-chown nobody:nogroup "$TARGET_DIR"
-chmod -R 777 "$TARGET_DIR"
 
 service netatalk restart
diff --git a/vimrc/.vim/bundle/vim-autoformat b/vimrc/.vim/bundle/vim-autoformat
index ab6402f1..bdc4ef18 160000
--- ./vimrc/.vim/bundle/vim-autoformat
+++ ./vimrc/.vim/bundle/vim-autoformat
@@ -1 +1 @@
-Subproject commit ab6402f1a5e2704c353ab8d3de6e333603005fd4
+Subproject commit bdc4ef1862f0e8b57d9fe7bfbfe6451ff814f20c
diff --git a/vimrc/.vim/bundle/vim-snipmate b/vimrc/.vim/bundle/vim-snipmate
index cbec960a..ed3c5426 160000
--- ./vimrc/.vim/bundle/vim-snipmate
+++ ./vimrc/.vim/bundle/vim-snipmate
@@ -1 +1 @@
-Subproject commit cbec960ab558b20281c0634b9b1a45fb16aaf638
+Subproject commit ed3c5426a20bf1c06d7980946ada34fd9f93320e
diff --git a/wifirc/autoconnect.sh b/wifirc/autoconnect.sh
index e8d338a2..430298e0 100755
--- ./wifirc/autoconnect.sh
+++ ./wifirc/autoconnect.sh
@@ -4,12 +4,16 @@
 # and put this script, which is owned by root and executable, in the folder: /etc/NetworkManager/dispatcher.d
 
 # the following command to get the wifi device:
-# for dev in `ls /sys/class/net`; do
-#     [ -d "/sys/class/net/$dev/wireless" ] && echo "$dev"
-# done
 
+DEV=$(
+    for dev in `ls /sys/class/net`; do
+        [ -d "/sys/class/net/$dev/wireless" ] && echo "$dev"
+    done
+) 
 
-if [ "$CONNECTION_UUID" = "1c853a09-cf79-4a54-86fa-9efb5e447ea8" ]
+TARGET_ID=$(nmcli connection show | grep $DEV | sed 's/.*[^0-9a-z]\(\([0-9a-z]*-\)\{4\}[0-9a-z]*\).*/\1/g')
+
+if [ "$CONNECTION_UUID" = "$TARGET_ID" ]
 then
     if [ "$2" = "down" ]; then
         sleep 10
